<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>COVID-19 Host Transcriptomic Analysis</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<!-- Meta -->
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">

<!-- SEO (optional but recommended) -->
<meta name="description" content="Ethan Saline Portfolio – Data Science, Machine Learning, Bioinformatics, and Statistical Analysis">
<meta name="author" content="Ethan Saline">

<!-- Google Font -->
<link href="https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&display=swap" rel="stylesheet">

<!-- Optional: smooth scrolling -->
<style>
  html {
    scroll-behavior: smooth;
  }
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><span style='font-family: Arial Black, Impact, sans-serif; font-weight: 600; font-size: 32px;'> Ethan Saline's Portfolio </span></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="lung_cnn.html">Lung X-Ray CNN</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Statistical Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="logistic_regression.html">Logistic Regression: Heart Failure</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Bioinformatics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="covid.html">Covid Transcriptomics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Storytelling
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ICE.html">Covid Transcriptomics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Time Series
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="flu.html">Influenza Decomposition</a>
    </li>
    <li>
      <a href="credit_cards.html">Credit Card Loans Decomposition</a>
    </li>
    <li>
      <a href="palantir.html">Palantir Stock Stocastic Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">COVID-19 Host Transcriptomic Analysis</h1>

</div>


<div id="executive-summary" class="section level2">
<h2>Executive Summary</h2>
<p>SARS-CoV-2 infection induces profound alterations in the host
transcriptome, characterized by upregulation of inflammatory and
interferon-stimulated genes and downregulation of epithelial and
regulatory genes. Using normalized RNA-seq datasets from multiple human
cell types and lung biopsies, we conducted a comprehensive analysis
combining differential expression, dimensionality reduction, and
targeted gene evaluation.</p>
<p><strong>Key insights include</strong>:</p>
<ul>
<li><strong>Distinct COVID-19 transcriptional signature</strong>:
COVID-19 samples display elevated expression of antiviral and immune
response genes (e.g., IRAK2, SOD2) and reduced expression of epithelial
maintenance genes (e.g., ANXA8, ANXA8L1).</li>
<li><strong>Cell-type specificity</strong>: NHBE and A549 cells exhibit
pronounced transcriptional shifts</li>
<li><strong>Co-expression and correlation</strong>: Selected genes show
strong pairwise correlations, indicating coordinated regulation of
immune and epithelial pathways.</li>
<li><strong>Validation through multiple approaches</strong>: PCA, t-SNE,
and UMAP reveal clear separation between infected and control samples,
confirming robust transcriptomic differences.</li>
</ul>
<p>Overall, the findings provide a molecular roadmap of host response to
SARS-CoV-2, highlighting potential targets for therapeutic intervention
and further functional studies.</p>
<hr />
<div class="float">
<img src="proj_files/covid.jpeg" alt="Covid Diagram" />
<div class="figcaption">Covid Diagram</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>SARS-CoV-2 infection triggers an atypical host transcriptomic
response characterized by <strong>weak type I/III interferon
signaling</strong> and <strong>strong chemokine/cytokine
induction</strong>.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a> This imbalance allows viral persistence
while promoting inflammation, tissue damage, and immune dysregulation
(Blanco-Melo et al., 2020). Understanding the transcriptional landscape
of infected tissues can reveal <strong>key genes, co-expression modules,
and pathways</strong> underlying these responses. This analysis aims to
provide actionable insights into the molecular mechanisms of COVID-19.<a
href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
<hr />
<div id="research-question" class="section level3">
<h3>Research Question</h3>
<p>Which genes distinguish COVID-19 patients from healthy controls, and
what do their expression patterns reveal about the host transcriptomic
response to SARS-CoV-2 infection?</p>
<p><strong>Hypotheses:</strong></p>
<ol style="list-style-type: decimal">
<li>Interferon-stimulated genes (ISGs) show altered expression in
COVID-19 samples</li>
<li>Inflammatory and chemokine genes are upregulated in infected
tissues</li>
<li>Expression patterns vary by cell type and infection status</li>
</ol>
<hr />
</div>
<div id="dataset-summary" class="section level3">
<h3>Dataset Summary</h3>
<table>
<colgroup>
<col width="20%" />
<col width="70%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th>Object Name</th>
<th>Description</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>expr_matrix</strong></td>
<td>Normalized RNA-seq counts (FPKM)</td>
<td>Gene-level expression</td>
</tr>
<tr class="even">
<td><strong>sample_info</strong></td>
<td>Sample metadata (COVID vs healthy, tissue type)</td>
<td>Includes batch/donor info</td>
</tr>
<tr class="odd">
<td><strong>gene_annot</strong></td>
<td>Gene annotation (Ensembl ID, symbol, type)</td>
<td>Mapping and filtering</td>
</tr>
</tbody>
</table>
<p><strong>Sample Summary</strong></p>
<table>
<colgroup>
<col width="37%" />
<col width="2%" />
<col width="15%" />
<col width="43%" />
</colgroup>
<thead>
<tr class="header">
<th>Group / Treatment</th>
<th>n</th>
<th>Cell Type</th>
<th>Virus / Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HPIV3 infected</td>
<td>3</td>
<td>A549</td>
<td>HPIV3</td>
</tr>
<tr class="even">
<td>Human IFNB treated</td>
<td>6</td>
<td>NHBE</td>
<td>IFNB (4h, 6h, 12h timepoints)</td>
</tr>
<tr class="odd">
<td>IAV infected</td>
<td>2</td>
<td>A549</td>
<td>Influenza A Virus (IAV)</td>
</tr>
<tr class="even">
<td>IAV infected</td>
<td>4</td>
<td>NHBE</td>
<td>Influenza A Virus (IAV)</td>
</tr>
<tr class="odd">
<td>IAVdNS1 infected</td>
<td>4</td>
<td>NHBE</td>
<td>Influenza A Virus ΔNS1</td>
</tr>
<tr class="even">
<td>Healthy control</td>
<td>2</td>
<td>Lung biopsy</td>
<td>None (negative control)</td>
</tr>
<tr class="odd">
<td>COVID-19 patient</td>
<td>2</td>
<td>Lung biopsy</td>
<td>SARS-CoV-2 (in vivo)</td>
</tr>
<tr class="even">
<td>Mock treated</td>
<td>13</td>
<td>A549</td>
<td>Mock</td>
</tr>
<tr class="odd">
<td>Mock treated + ACE2</td>
<td>9</td>
<td>A549</td>
<td>Mock + ACE2 expression</td>
</tr>
<tr class="even">
<td>Mock treated</td>
<td>3</td>
<td>Calu-3</td>
<td>Mock</td>
</tr>
<tr class="odd">
<td>Mock treated</td>
<td>7</td>
<td>NHBE</td>
<td>Mock</td>
</tr>
<tr class="even">
<td>RSV infected</td>
<td>5</td>
<td>A549</td>
<td>RSV</td>
</tr>
<tr class="odd">
<td>SARS-CoV-2 infected</td>
<td>6</td>
<td>A549</td>
<td>SARS-CoV-2</td>
</tr>
<tr class="even">
<td>SARS-CoV-2 infected</td>
<td>6</td>
<td>A549 + ACE2</td>
<td>SARS-CoV-2</td>
</tr>
<tr class="odd">
<td>SARS-CoV-2 infected</td>
<td>3</td>
<td>Calu-3</td>
<td>SARS-CoV-2</td>
</tr>
<tr class="even">
<td>SARS-CoV-2 infected</td>
<td>3</td>
<td>NHBE</td>
<td>SARS-CoV-2</td>
</tr>
<tr class="odd">
<td>SARS-CoV-2 + Ruxolitinib</td>
<td>3</td>
<td>A549 + ACE2</td>
<td>SARS-CoV-2 + Ruxolitinib</td>
</tr>
<tr class="even">
<td><strong>Total</strong></td>
<td><strong>78</strong></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<hr />
</div>
<div id="analytical-pipeline" class="section level3">
<h3>Analytical Pipeline</h3>
<div
style="display:flex;flex-direction:column;align-items:center;font-family:Arial,sans-serif;">
<div
style="background-color:#2C3E50;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Data Loading &amp; QC</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#16A085;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Preprocessing &amp; Normalization</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#2980B9;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Exploratory Analysis (PCA / UMAP)</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#D35400;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Batch Assessment &amp; Model Design</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#E67E22;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Differential Expression</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#8E44AD;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Co-expression Modules (WGCNA)</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#F1C40F;color:black;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Functional Enrichment (GO/KEGG)</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#27AE60;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Integration &amp; Interpretation</p>
</div>
<div style="margin:5px;font-size:20px;">
↓
</div>
<div
style="background-color:#34495E;color:white;padding:10px 15px;margin:3px;border-radius:8px;width:260px;text-align:center;">
<p>Summary &amp; Reporting</p>
</div>
</div>
<hr />
</div>
</div>
<div id="data-preparation" class="section level2">
<h2>Data Preparation</h2>
<div id="data-loading-quality-control" class="section level3">
<h3>Data Loading &amp; Quality Control</h3>
<div id="package-installation-and-loading" class="section level4">
<h4>Package Installation and Loading</h4>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Install and load required packages</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pacman&quot;</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;pacman&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  tidyverse,    <span class="co"># Data manipulation and visualization</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  limma,        <span class="co"># Linear models for microarray/RNA-seq</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  edgeR,        <span class="co"># Differential expression analysis</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  DESeq2,       <span class="co"># Differential expression (negative binomial)</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  WGCNA,        <span class="co"># Weighted gene co-expression network analysis</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>  pheatmap,     <span class="co"># Pretty heatmaps</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  Rtsne,        <span class="co"># t-SNE dimensionality reduction</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>  uwot,         <span class="co"># UMAP dimensionality reduction</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>  ggrepel,      <span class="co"># Non-overlapping text labels</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>  kableExtra,   <span class="co"># Enhanced tables</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>  pander        <span class="co"># Formatted output</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="import-datasets" class="section level4">
<h4>Import Datasets</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Define file paths</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">&quot;data&quot;</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">expr =</span> <span class="st">&quot;GSE147507_norm_counts_FPKM_GRCh38.p13_NCBI.tsv&quot;</span>,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">meta =</span> <span class="st">&quot;GSE147507-GPL18573_series_matrix.tsv&quot;</span>,</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">annot =</span> <span class="st">&quot;Human.GRCh38.p13.annot.tsv&quot;</span>,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">counts =</span> <span class="st">&quot;GSE147507_RawReadCounts_Human.tsv&quot;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co"># Load all datasets</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>expr_matrix <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir, files<span class="sc">$</span>expr), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir, files<span class="sc">$</span>meta), <span class="at">skip =</span> <span class="dv">1</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>gene_annot <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir, files<span class="sc">$</span>annot), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>raw_counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir, files<span class="sc">$</span>counts), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## New names:
## • `` -&gt; `...1`</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Report dimensions</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Datasets successfully loaded</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Datasets successfully loaded</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Expression matrix: %d genes × %d samples</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>            <span class="fu">nrow</span>(expr_matrix), <span class="fu">ncol</span>(expr_matrix) <span class="sc">-</span> <span class="dv">1</span>))</span></code></pre></div>
<pre><code>## Expression matrix: 39376 genes × 67 samples</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Sample metadata: %d rows × %d columns</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>            <span class="fu">nrow</span>(sample_info), <span class="fu">ncol</span>(sample_info)))</span></code></pre></div>
<pre><code>## Sample metadata: 46 rows × 79 columns</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Gene annotation: %d genes annotated</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>            <span class="fu">nrow</span>(gene_annot)))</span></code></pre></div>
<pre><code>## Gene annotation: 39376 genes annotated</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Raw counts: %d genes × %d samples</span><span class="sc">\n\n</span><span class="st">&quot;</span>, </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>            <span class="fu">nrow</span>(raw_counts), <span class="fu">ncol</span>(raw_counts) <span class="sc">-</span> <span class="dv">1</span>))</span></code></pre></div>
<pre><code>## Raw counts: 21797 genes × 78 samples</code></pre>
<hr />
</div>
</div>
<div id="quality-control" class="section level3">
<h3>Quality Control</h3>
<div id="sample-alignment-verification" class="section level4">
<h4>Sample Alignment Verification</h4>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># Extract sample names from expression matrix (excluding GeneID column)</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>expr_samples <span class="ot">&lt;-</span> <span class="fu">colnames</span>(expr_matrix)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co"># Find common samples between datasets</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_samples, <span class="fu">colnames</span>(sample_info))</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a><span class="co"># Report alignment statistics</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Sample Alignment Summary:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Sample Alignment Summary:</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Expression matrix: %d samples</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">length</span>(expr_samples)))</span></code></pre></div>
<pre><code>##   Expression matrix: 67 samples</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Metadata: %d sample columns</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">ncol</span>(sample_info)))</span></code></pre></div>
<pre><code>##   Metadata: 79 sample columns</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Common samples: %d</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">length</span>(common_samples)))</span></code></pre></div>
<pre><code>##   Common samples: 67</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Subset metadata to match expression data</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>sample_info_sub <span class="ot">&lt;-</span> sample_info[, <span class="fu">colnames</span>(sample_info) <span class="sc">%in%</span> expr_samples]</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Filtered metadata dimensions: %d rows × %d columns</span><span class="sc">\n\n</span><span class="st">&quot;</span>, </span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>            <span class="fu">nrow</span>(sample_info_sub), <span class="fu">ncol</span>(sample_info_sub)))</span></code></pre></div>
<pre><code>## Filtered metadata dimensions: 46 rows × 67 columns</code></pre>
</div>
<div id="missing-value-check" class="section level4">
<h4>Missing Value Check</h4>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co"># Check for missing values in expression data</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>n_missing <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(expr_matrix))</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>pct_missing <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="sc">*</span> n_missing <span class="sc">/</span> <span class="fu">prod</span>(<span class="fu">dim</span>(expr_matrix))</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Missing Value Analysis:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Missing Value Analysis:</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Total missing: %d (%.2f%%)</span><span class="sc">\n</span><span class="st">&quot;</span>, n_missing, pct_missing))</span></code></pre></div>
<pre><code>##   Total missing: 0 (0.00%)</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="cf">if</span> (n_missing <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;  ⚠ Consider imputation or removal of affected genes/samples</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;  ✓ No missing values detected</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>##   ✓ No missing values detected</code></pre>
<hr />
</div>
</div>
<div id="data-preprocessing" class="section level3">
<h3>Data Preprocessing</h3>
<div id="create-annotation-mapping" class="section level4">
<h4>Create Annotation Mapping</h4>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="co"># Create sample annotation dataframe</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>sample_groups <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(sample_info_sub[<span class="dv">6</span>, <span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>sample_annot <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(sample_info_sub)[<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  <span class="at">group_full =</span> sample_groups</span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>  <span class="co"># Standardize group names</span></span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;Mock&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Mock&quot;</span>,</span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;SARS-CoV-2&quot;</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">str_detect</span>(group_full, <span class="st">&quot;Rux&quot;</span>) <span class="sc">~</span> <span class="st">&quot;SARS2&quot;</span>,</span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;RSV&quot;</span>) <span class="sc">~</span> <span class="st">&quot;RSV&quot;</span>,</span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;IAV&quot;</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">str_detect</span>(group_full, <span class="st">&quot;dNS1&quot;</span>) <span class="sc">~</span> <span class="st">&quot;IAV&quot;</span>,</span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;IAVdNS1&quot;</span>) <span class="sc">~</span> <span class="st">&quot;IAVdNS1&quot;</span>,</span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;HPIV3&quot;</span>) <span class="sc">~</span> <span class="st">&quot;HPIV3&quot;</span>,</span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;IFNB&quot;</span>) <span class="sc">~</span> <span class="st">&quot;IFNB&quot;</span>,</span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;Lung.*healthy&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Healthy&quot;</span>,</span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;COVID&quot;</span>) <span class="sc">~</span> <span class="st">&quot;COVID19&quot;</span>,</span>
<span id="cb30-19"><a href="#cb30-19" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span></span>
<span id="cb30-20"><a href="#cb30-20" tabindex="-1"></a>    ),</span>
<span id="cb30-21"><a href="#cb30-21" tabindex="-1"></a>    <span class="co"># Extract cell type</span></span>
<span id="cb30-22"><a href="#cb30-22" tabindex="-1"></a>    <span class="at">cell_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-23"><a href="#cb30-23" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;NHBE&quot;</span>) <span class="sc">~</span> <span class="st">&quot;NHBE&quot;</span>,</span>
<span id="cb30-24"><a href="#cb30-24" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;A549.*ACE2&quot;</span>) <span class="sc">~</span> <span class="st">&quot;A549_ACE2&quot;</span>,</span>
<span id="cb30-25"><a href="#cb30-25" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;A549&quot;</span>) <span class="sc">~</span> <span class="st">&quot;A549&quot;</span>,</span>
<span id="cb30-26"><a href="#cb30-26" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;Calu-3&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Calu3&quot;</span>,</span>
<span id="cb30-27"><a href="#cb30-27" tabindex="-1"></a>      <span class="fu">str_detect</span>(group_full, <span class="st">&quot;Lung&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Lung&quot;</span>,</span>
<span id="cb30-28"><a href="#cb30-28" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Unknown&quot;</span></span>
<span id="cb30-29"><a href="#cb30-29" tabindex="-1"></a>    )</span>
<span id="cb30-30"><a href="#cb30-30" tabindex="-1"></a>  )</span>
<span id="cb30-31"><a href="#cb30-31" tabindex="-1"></a></span>
<span id="cb30-32"><a href="#cb30-32" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Sample annotation created:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Sample annotation created:</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">table</span>(sample_annot<span class="sc">$</span>group, sample_annot<span class="sc">$</span>cell_type) <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  <span class="fu">addmargins</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="st">&quot;condensed&quot;</span>)</span></code></pre></div>
<table class="table table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
A549
</th>
<th style="text-align:right;">
A549_ACE2
</th>
<th style="text-align:right;">
Calu3
</th>
<th style="text-align:right;">
Lung
</th>
<th style="text-align:right;">
NHBE
</th>
<th style="text-align:right;">
Sum
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
HPIV3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
IAV
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
IAVdNS1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
IFNB
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
Mock
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
28
</td>
</tr>
<tr>
<td style="text-align:left;">
Other
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RSV
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
SARS2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
Sum
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
66
</td>
</tr>
</tbody>
</table>
</div>
<div id="log-transform-expression-data" class="section level4">
<h4>Log-Transform Expression Data</h4>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># Extract numeric expression values and log-transform</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>fpkm_mat <span class="ot">&lt;-</span> expr_matrix <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a><span class="co"># Add gene IDs as rownames</span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a><span class="fu">rownames</span>(fpkm_mat) <span class="ot">&lt;-</span> expr_matrix<span class="sc">$</span>GeneID</span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a></span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a><span class="co"># Log2 transformation with pseudocount</span></span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>log_mat <span class="ot">&lt;-</span> <span class="fu">log2</span>(fpkm_mat <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a><span class="co"># Remove zero-variance genes for downstream analysis</span></span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>gene_vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(log_mat, <span class="dv">1</span>, var, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>log_mat_filt <span class="ot">&lt;-</span> log_mat[gene_vars <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Expression matrix after filtering:</span><span class="sc">\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<pre><code>## Expression matrix after filtering:</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  %d genes × %d samples</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">nrow</span>(log_mat_filt), <span class="fu">ncol</span>(log_mat_filt)))</span></code></pre></div>
<pre><code>##   35457 genes × 67 samples</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  %d zero-variance genes removed</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">sum</span>(gene_vars <span class="sc">==</span> <span class="dv">0</span>)))</span></code></pre></div>
<pre><code>##   3919 zero-variance genes removed</code></pre>
<hr />
</div>
</div>
</div>
<div id="exploratory-analysis" class="section level2">
<h2>Exploratory Analysis</h2>
<div id="expression-distribution-visualization" class="section level3">
<h3>Expression Distribution Visualization</h3>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="co"># Prepare long-format data for plotting</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> log_mat <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>gene,</span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;expression&quot;</span></span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_annot, <span class="at">by =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" tabindex="-1"></a><span class="co"># Boxplot by condition</span></span>
<span id="cb39-13"><a href="#cb39-13" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> expression, <span class="at">fill =</span> group)) <span class="sc">+</span></span>
<span id="cb39-14"><a href="#cb39-14" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.size =</span> <span class="fl">0.3</span>, <span class="at">outlier.alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb39-15"><a href="#cb39-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb39-16"><a href="#cb39-16" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb39-17"><a href="#cb39-17" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb39-18"><a href="#cb39-18" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb39-19"><a href="#cb39-19" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb39-20"><a href="#cb39-20" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb39-21"><a href="#cb39-21" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Expression Distribution by Condition&quot;</span>,</span>
<span id="cb39-22"><a href="#cb39-22" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb39-23"><a href="#cb39-23" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;log2(FPKM + 1)&quot;</span></span>
<span id="cb39-24"><a href="#cb39-24" tabindex="-1"></a>  )</span>
<span id="cb39-25"><a href="#cb39-25" tabindex="-1"></a></span>
<span id="cb39-26"><a href="#cb39-26" tabindex="-1"></a><span class="co"># Density plot by condition</span></span>
<span id="cb39-27"><a href="#cb39-27" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> expression, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb39-28"><a href="#cb39-28" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb39-29"><a href="#cb39-29" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb39-30"><a href="#cb39-30" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb39-31"><a href="#cb39-31" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb39-32"><a href="#cb39-32" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Expression Density by Condition&quot;</span>,</span>
<span id="cb39-33"><a href="#cb39-33" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;log2(FPKM + 1)&quot;</span>,</span>
<span id="cb39-34"><a href="#cb39-34" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Density&quot;</span>,</span>
<span id="cb39-35"><a href="#cb39-35" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Condition&quot;</span></span>
<span id="cb39-36"><a href="#cb39-36" tabindex="-1"></a>  )</span>
<span id="cb39-37"><a href="#cb39-37" tabindex="-1"></a></span>
<span id="cb39-38"><a href="#cb39-38" tabindex="-1"></a><span class="co"># Display plots</span></span>
<span id="cb39-39"><a href="#cb39-39" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="covid_files/figure-html/viz_distributions-1.png" width="960" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>p2</span></code></pre></div>
<p><img src="covid_files/figure-html/viz_distributions-2.png" width="960" /></p>
<p>A density plot visualizes the distribution of gene expression values
across samples. Each line represents one sample or one group of samples.
Instead of using bars like a histogram, density curves show a smooth
estimate of how gene expression values are distributed.</p>
<p>If curves from all samples overlap closely, it means: - expression
values were successfully normalized, - there is no severe batch effect,
and - no sample has extreme or unusual distribution.</p>
</div>
<div id="principal-component-analysis" class="section level3">
<h3>Principal Component Analysis</h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># Perform PCA on filtered log-transformed data</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>pca_res <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(log_mat_filt), <span class="at">scale. =</span> <span class="cn">TRUE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a><span class="co"># Calculate variance explained</span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>pve <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="sc">*</span> pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a><span class="co"># Create PCA dataframe</span></span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_res<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_annot, <span class="at">by =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a><span class="fu">ggplot</span>(pca_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;PCA: Expression Patterns by Cell Type and Condition&quot;</span>,</span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;All together&quot;</span>,</span>
<span id="cb41-18"><a href="#cb41-18" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">sprintf</span>(<span class="st">&quot;PC1 (%.1f%% variance)&quot;</span>, pve[<span class="dv">1</span>]),</span>
<span id="cb41-19"><a href="#cb41-19" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">sprintf</span>(<span class="st">&quot;PC2 (%.1f%% variance)&quot;</span>, pve[<span class="dv">2</span>]),</span>
<span id="cb41-20"><a href="#cb41-20" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Condition&quot;</span></span>
<span id="cb41-21"><a href="#cb41-21" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-22"><a href="#cb41-22" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="covid_files/figure-html/viz_pca-1.png" width="960" /></p>
<p>The PCA plots shows how samples group together based on their overall
gene expression</p>
<p>PCA reduces thousands of genes into a few ‘principal components’ that
capture the major sources of variation in the dataset. Each point is a
sample, positioned based on its overall gene expression profile. We see
clear seperation based on cell type.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="co"># PCA plot faceted by cell type</span></span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="fu">ggplot</span>(pca_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> cell_type, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;PCA: Expression Patterns by Cell Type and Condition&quot;</span>,</span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;Cell type is the primary driver of variation&quot;</span>,</span>
<span id="cb42-9"><a href="#cb42-9" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">sprintf</span>(<span class="st">&quot;PC1 (%.1f%% variance)&quot;</span>, pve[<span class="dv">1</span>]),</span>
<span id="cb42-10"><a href="#cb42-10" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">sprintf</span>(<span class="st">&quot;PC2 (%.1f%% variance)&quot;</span>, pve[<span class="dv">2</span>]),</span>
<span id="cb42-11"><a href="#cb42-11" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Condition&quot;</span></span>
<span id="cb42-12"><a href="#cb42-12" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb42-13"><a href="#cb42-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="covid_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p>Once facetted by cell type we gain more insight.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="co"># Scree plot</span></span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">tibble</span>(<span class="at">PC =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">Variance =</span> pve[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]), </span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y =</span> Variance)) <span class="sc">+</span></span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;#2C3E50&quot;</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">&quot;#E74C3C&quot;</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Scree Plot&quot;</span>,</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;Variance Explained by Principal Components&quot;</span>,</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Principal Component&quot;</span>,</span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;% Variance Explained&quot;</span></span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="covid_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>The scree plot shows how much variation each principal component
captures. The first Point represents PC1, the second PC2, and so on.</p>
<p>We see a steep drop after the first few, <strong>meaning the first
few components capture most of the structure in the data</strong>.</p>
<p>When the line flattens out, later components mostly represent
noise.</p>
<p>That tells us <strong>the above PCA visualizations are
meaningful</strong>, and that only the first components reflect real
biological structure, while later ones represent only noise</p>
</div>
<div id="t-sne-dimensionality-reduction" class="section level3">
<h3>t-SNE Dimensionality Reduction</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="co"># Use first 50 PCs for t-SNE</span></span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>tsne_res <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(</span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a>  pca_res<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],</span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>  <span class="at">dims =</span> <span class="dv">2</span>,</span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>  <span class="at">perplexity =</span> <span class="dv">10</span>,</span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>)</span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a></span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a><span class="co"># Create t-SNE dataframe</span></span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>tsne_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a>  <span class="at">TSNE1 =</span> tsne_res<span class="sc">$</span>Y[, <span class="dv">1</span>],</span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a>  <span class="at">TSNE2 =</span> tsne_res<span class="sc">$</span>Y[, <span class="dv">2</span>],</span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">rownames</span>(pca_res<span class="sc">$</span>x)</span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_annot, <span class="at">by =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a></span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a><span class="co"># t-SNE plot</span></span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a><span class="fu">ggplot</span>(tsne_df, <span class="fu">aes</span>(<span class="at">x =</span> TSNE1, <span class="at">y =</span> TSNE2, <span class="at">color =</span> group, <span class="at">shape =</span> cell_type)) <span class="sc">+</span></span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;t-SNE Visualization of Sample Relationships&quot;</span>,</span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;Upper cluster: NHBE and A549 cells | Lower right: Calu-3 cells&quot;</span>,</span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Condition&quot;</span>,</span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">&quot;Cell Type&quot;</span></span>
<span id="cb44-27"><a href="#cb44-27" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb44-28"><a href="#cb44-28" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<p><img src="covid_files/figure-html/viz_tsne-1.png" width="768" /></p>
<p>t-SNE is a nonlinear dimensionality reduction method designed to
preserve local structure, <strong>meaning it keeps similar samples
grouped together</strong>.</p>
<p>How to interpret it - <strong>Tight clusters</strong> =&gt; samples
with very similar expression - <strong>Large distances</strong> =&gt;
major biological differences</p>
<p>t-SNE emphasizes neighborhood relationships; the global layout
(overall shape) is less meaningful.</p>
</div>
<div id="umap-dimensionality-reduction" class="section level3">
<h3>UMAP Dimensionality Reduction</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="co"># Perform UMAP</span></span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>umap_res <span class="ot">&lt;-</span> <span class="fu">umap</span>(</span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>  pca_res<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],</span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>  <span class="at">n_neighbors =</span> <span class="dv">15</span>,</span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a>  <span class="at">min_dist =</span> <span class="fl">0.1</span>,</span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>  <span class="at">metric =</span> <span class="st">&quot;correlation&quot;</span></span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a>)</span>
<span id="cb45-9"><a href="#cb45-9" tabindex="-1"></a></span>
<span id="cb45-10"><a href="#cb45-10" tabindex="-1"></a><span class="co"># Create UMAP dataframe</span></span>
<span id="cb45-11"><a href="#cb45-11" tabindex="-1"></a>umap_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb45-12"><a href="#cb45-12" tabindex="-1"></a>  <span class="at">UMAP1 =</span> umap_res[, <span class="dv">1</span>],</span>
<span id="cb45-13"><a href="#cb45-13" tabindex="-1"></a>  <span class="at">UMAP2 =</span> umap_res[, <span class="dv">2</span>],</span>
<span id="cb45-14"><a href="#cb45-14" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">rownames</span>(pca_res<span class="sc">$</span>x)</span>
<span id="cb45-15"><a href="#cb45-15" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb45-16"><a href="#cb45-16" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_annot, <span class="at">by =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb45-17"><a href="#cb45-17" tabindex="-1"></a></span>
<span id="cb45-18"><a href="#cb45-18" tabindex="-1"></a><span class="co"># UMAP plot</span></span>
<span id="cb45-19"><a href="#cb45-19" tabindex="-1"></a><span class="fu">ggplot</span>(umap_df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP1, <span class="at">y =</span> UMAP2, <span class="at">color =</span> group, <span class="at">shape =</span> cell_type)) <span class="sc">+</span></span>
<span id="cb45-20"><a href="#cb45-20" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb45-21"><a href="#cb45-21" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb45-22"><a href="#cb45-22" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb45-23"><a href="#cb45-23" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;UMAP Visualization of Sample Relationships&quot;</span>,</span>
<span id="cb45-24"><a href="#cb45-24" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;Clear cell type clustering: A549 (left) | NHBE (center) | Calu-3 (right)&quot;</span>,</span>
<span id="cb45-25"><a href="#cb45-25" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Condition&quot;</span>,</span>
<span id="cb45-26"><a href="#cb45-26" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">&quot;Cell Type&quot;</span></span>
<span id="cb45-27"><a href="#cb45-27" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb45-28"><a href="#cb45-28" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<p><img src="covid_files/figure-html/viz_umap-1.png" width="768" /></p>
<p>Umap is similar to t-SNE but better at preserving both local and
global structure. <strong>It tends to form more meaningful cluster
shapes</strong></p>
<p>This UMAP plot shows clusters of samples grouped by their gene
expression profiles. We again see clear separation between each kind of
infected and healthy samples, reinforcing the idea that COVID-19 causes
strong transcriptomic shifts</p>
<hr />
</div>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2>Differential Expression Analysis</h2>
<p>DESeq2 is a statistical method used to <strong>identify genes that
are significantly differentially expressed between two
conditions</strong>: in our case, COVID-19 versus healthy controls.</p>
<p><em>NOTE</em> We used DESeq2 instead of limma because DESeq2 was
specifically designed for raw RNA-seq count data. It uses a negative
binomial model, which correctly captures the distribution and
variability of RNA-seq counts. DESeq2 also provides strong dispersion
estimation and performs better with small sample sizes, which is
important for COVID-19 datasets. Limma is excellent for microarrays or
large, normalized datasets, but <strong>DESeq2 gives more robust and
accurate differential expression results for count-based
RNA-seq</strong>, which is why we selected it for this analysis</p>
<div id="prepare-deseq2-input" class="section level3">
<h3>Prepare DESeq2 Input</h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="co"># Extract gene IDs and create count matrix</span></span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> raw_counts[[<span class="dv">1</span>]]</span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(raw_counts[, <span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a><span class="fu">rownames</span>(count_mat) <span class="ot">&lt;-</span> gene_ids</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a><span class="co"># Create condition labels from sample names</span></span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>condition_map <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(count_mat)</span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">case_when</span>(</span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;SARS-CoV-2&quot;</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">str_detect</span>(sample, <span class="st">&quot;Rux&quot;</span>) <span class="sc">~</span> <span class="st">&quot;COVID19&quot;</span>,</span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;Mock|HealthyLung&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Control&quot;</span>,</span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span></span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a>    ),</span>
<span id="cb46-16"><a href="#cb46-16" tabindex="-1"></a>    <span class="at">batch =</span> <span class="fu">case_when</span>(</span>
<span id="cb46-17"><a href="#cb46-17" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;A549-ACE2&quot;</span>) <span class="sc">~</span> <span class="st">&quot;A549_ACE2&quot;</span>,</span>
<span id="cb46-18"><a href="#cb46-18" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;A549&quot;</span>) <span class="sc">~</span> <span class="st">&quot;A549&quot;</span>,</span>
<span id="cb46-19"><a href="#cb46-19" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;NHBE&quot;</span>) <span class="sc">~</span> <span class="st">&quot;NHBE&quot;</span>,</span>
<span id="cb46-20"><a href="#cb46-20" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;Calu3&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Calu3&quot;</span>,</span>
<span id="cb46-21"><a href="#cb46-21" tabindex="-1"></a>      <span class="fu">str_detect</span>(sample, <span class="st">&quot;Lung&quot;</span>) <span class="sc">~</span> <span class="st">&quot;Lung&quot;</span>,</span>
<span id="cb46-22"><a href="#cb46-22" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span></span>
<span id="cb46-23"><a href="#cb46-23" tabindex="-1"></a>    )</span>
<span id="cb46-24"><a href="#cb46-24" tabindex="-1"></a>  )</span>
<span id="cb46-25"><a href="#cb46-25" tabindex="-1"></a></span>
<span id="cb46-26"><a href="#cb46-26" tabindex="-1"></a><span class="co"># Filter to COVID-19 vs Control comparison</span></span>
<span id="cb46-27"><a href="#cb46-27" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> condition_map <span class="sc">%&gt;%</span></span>
<span id="cb46-28"><a href="#cb46-28" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;COVID19&quot;</span>, <span class="st">&quot;Control&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-29"><a href="#cb46-29" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb46-30"><a href="#cb46-30" tabindex="-1"></a></span>
<span id="cb46-31"><a href="#cb46-31" tabindex="-1"></a><span class="co"># Subset count matrix to match</span></span>
<span id="cb46-32"><a href="#cb46-32" tabindex="-1"></a>count_mat_filt <span class="ot">&lt;-</span> count_mat[, <span class="fu">rownames</span>(coldata)]</span>
<span id="cb46-33"><a href="#cb46-33" tabindex="-1"></a></span>
<span id="cb46-34"><a href="#cb46-34" tabindex="-1"></a><span class="co"># Assess batch confounding</span></span>
<span id="cb46-35"><a href="#cb46-35" tabindex="-1"></a>batch_table <span class="ot">&lt;-</span> <span class="fu">table</span>(coldata<span class="sc">$</span>batch, coldata<span class="sc">$</span>condition)</span>
<span id="cb46-36"><a href="#cb46-36" tabindex="-1"></a>valid_batches <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">which</span>(<span class="fu">rowSums</span>(batch_table <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">==</span> <span class="dv">2</span>))</span>
<span id="cb46-37"><a href="#cb46-37" tabindex="-1"></a></span>
<span id="cb46-38"><a href="#cb46-38" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Batch Distribution for COVID-19 vs Control:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Batch Distribution for COVID-19 vs Control:</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="fu">addmargins</span>(batch_table) <span class="sc">%&gt;%</span> <span class="fu">kable</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Control
</th>
<th style="text-align:right;">
COVID19
</th>
<th style="text-align:right;">
Sum
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A549
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
A549_ACE2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
Calu3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
Lung
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
NHBE
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
Sum
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
49
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a><span class="co"># Determine if batch correction is feasible</span></span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a>use_batch <span class="ot">&lt;-</span> <span class="fu">length</span>(valid_batches) <span class="sc">&gt;=</span> <span class="dv">2</span></span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a><span class="cf">if</span> (use_batch) {</span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>  coldata <span class="ot">&lt;-</span> coldata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(batch <span class="sc">%in%</span> valid_batches)</span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>  count_mat_filt <span class="ot">&lt;-</span> count_mat_filt[, <span class="fu">rownames</span>(coldata)]</span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">✓ Using batch correction with batches: %s</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a>              <span class="fu">paste</span>(valid_batches, <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>)))</span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">⚠ Insufficient batch diversity - proceeding without batch correction</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## 
## ✓ Using batch correction with batches: A549, A549_ACE2, Calu3, NHBE</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a><span class="co"># Filter low-count genes (≥10 reads in ≥3 samples)</span></span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(count_mat_filt <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>count_mat_filt <span class="ot">&lt;-</span> count_mat_filt[keep_genes, ]</span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Final dataset dimensions: %d genes × %d samples</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a>            <span class="fu">nrow</span>(count_mat_filt), <span class="fu">ncol</span>(count_mat_filt)))</span></code></pre></div>
<pre><code>## 
## Final dataset dimensions: 14903 genes × 47 samples</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  COVID-19: %d samples</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">sum</span>(coldata<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">&quot;COVID19&quot;</span>)))</span></code></pre></div>
<pre><code>##   COVID-19: 18 samples</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Control: %d samples</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">sum</span>(coldata<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>)))</span></code></pre></div>
<pre><code>##   Control: 29 samples</code></pre>
</div>
<div id="run-deseq2" class="section level3">
<h3>Run DESeq2</h3>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a><span class="co"># Create DESeq2 object with appropriate design</span></span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a>design_formula <span class="ot">&lt;-</span> <span class="cf">if</span> (use_batch) {</span>
<span id="cb57-3"><a href="#cb57-3" tabindex="-1"></a>  <span class="sc">~</span> batch <span class="sc">+</span> condition</span>
<span id="cb57-4"><a href="#cb57-4" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb57-5"><a href="#cb57-5" tabindex="-1"></a>  <span class="sc">~</span> condition</span>
<span id="cb57-6"><a href="#cb57-6" tabindex="-1"></a>}</span>
<span id="cb57-7"><a href="#cb57-7" tabindex="-1"></a></span>
<span id="cb57-8"><a href="#cb57-8" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb57-9"><a href="#cb57-9" tabindex="-1"></a>  <span class="at">countData =</span> count_mat_filt,</span>
<span id="cb57-10"><a href="#cb57-10" tabindex="-1"></a>  <span class="at">colData =</span> coldata,</span>
<span id="cb57-11"><a href="#cb57-11" tabindex="-1"></a>  <span class="at">design =</span> design_formula</span>
<span id="cb57-12"><a href="#cb57-12" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a><span class="co"># Set reference level</span></span>
<span id="cb59-2"><a href="#cb59-2" tabindex="-1"></a>dds<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(dds<span class="sc">$</span>condition), <span class="at">ref =</span> <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb59-3"><a href="#cb59-3" tabindex="-1"></a></span>
<span id="cb59-4"><a href="#cb59-4" tabindex="-1"></a><span class="co"># Run differential expression analysis</span></span>
<span id="cb59-5"><a href="#cb59-5" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 69 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a><span class="co"># Extract results</span></span>
<span id="cb69-2"><a href="#cb69-2" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;COVID19&quot;</span>, <span class="st">&quot;Control&quot;</span>))</span>
<span id="cb69-3"><a href="#cb69-3" tabindex="-1"></a></span>
<span id="cb69-4"><a href="#cb69-4" tabindex="-1"></a><span class="co"># Summary</span></span>
<span id="cb69-5"><a href="#cb69-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;DESeq2 Analysis Complete</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## DESeq2 Analysis Complete</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">strrep</span>(<span class="st">&quot;=&quot;</span>, <span class="dv">50</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ==================================================</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>## 
## out of 14903 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 3400, 23%
## LFC &lt; 0 (down)     : 3327, 22%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 1)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<div id="filter-and-annotate-degs" class="section level3">
<h3>Filter and Annotate DEGs</h3>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" tabindex="-1"></a><span class="co"># Define significance thresholds</span></span>
<span id="cb76-2"><a href="#cb76-2" tabindex="-1"></a>padj_thresh <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb76-3"><a href="#cb76-3" tabindex="-1"></a>lfc_thresh <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb76-4"><a href="#cb76-4" tabindex="-1"></a></span>
<span id="cb76-5"><a href="#cb76-5" tabindex="-1"></a><span class="co"># Extract significant DEGs</span></span>
<span id="cb76-6"><a href="#cb76-6" tabindex="-1"></a>sig_degs <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb76-7"><a href="#cb76-7" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-8"><a href="#cb76-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;GENEID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-9"><a href="#cb76-9" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> padj_thresh, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc_thresh) <span class="sc">%&gt;%</span></span>
<span id="cb76-10"><a href="#cb76-10" tabindex="-1"></a>  <span class="fu">arrange</span>(padj)</span>
<span id="cb76-11"><a href="#cb76-11" tabindex="-1"></a></span>
<span id="cb76-12"><a href="#cb76-12" tabindex="-1"></a><span class="co"># Count directional changes</span></span>
<span id="cb76-13"><a href="#cb76-13" tabindex="-1"></a>n_up <span class="ot">&lt;-</span> <span class="fu">sum</span>(sig_degs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> lfc_thresh)</span>
<span id="cb76-14"><a href="#cb76-14" tabindex="-1"></a>n_down <span class="ot">&lt;-</span> <span class="fu">sum</span>(sig_degs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>lfc_thresh)</span>
<span id="cb76-15"><a href="#cb76-15" tabindex="-1"></a></span>
<span id="cb76-16"><a href="#cb76-16" tabindex="-1"></a><span class="co"># Annotate with gene symbols</span></span>
<span id="cb76-17"><a href="#cb76-17" tabindex="-1"></a>gene_annot_clean <span class="ot">&lt;-</span> gene_annot <span class="sc">%&gt;%</span></span>
<span id="cb76-18"><a href="#cb76-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneID =</span> <span class="fu">as.character</span>(GeneID)) <span class="sc">%&gt;%</span></span>
<span id="cb76-19"><a href="#cb76-19" tabindex="-1"></a>  <span class="fu">select</span>(GeneID, Symbol, Description)</span>
<span id="cb76-20"><a href="#cb76-20" tabindex="-1"></a></span>
<span id="cb76-21"><a href="#cb76-21" tabindex="-1"></a>sig_degs <span class="ot">&lt;-</span> sig_degs <span class="sc">%&gt;%</span></span>
<span id="cb76-22"><a href="#cb76-22" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_annot_clean, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;GENEID&quot;</span> <span class="ot">=</span> <span class="st">&quot;GeneID&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-23"><a href="#cb76-23" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Symbol =</span> <span class="fu">coalesce</span>(Symbol, GENEID))  <span class="co"># Use GENEID if Symbol missing</span></span>
<span id="cb76-24"><a href="#cb76-24" tabindex="-1"></a></span>
<span id="cb76-25"><a href="#cb76-25" tabindex="-1"></a><span class="co"># Report results</span></span>
<span id="cb76-26"><a href="#cb76-26" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">**Differential Expression Results**</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## **Differential Expression Results**</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Significant DEGs (FDR &lt; %.2f, |log2FC| &gt; %.1f): %d</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb78-2"><a href="#cb78-2" tabindex="-1"></a>            padj_thresh, lfc_thresh, <span class="fu">nrow</span>(sig_degs)))</span></code></pre></div>
<pre><code>##   Significant DEGs (FDR &lt; 0.05, |log2FC| &gt; 1.0): 775</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  ↑ Upregulated in COVID-19: %d</span><span class="sc">\n</span><span class="st">&quot;</span>, n_up))</span></code></pre></div>
<pre><code>##   ↑ Upregulated in COVID-19: 498</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  ↓ Downregulated in COVID-19: %d</span><span class="sc">\n\n</span><span class="st">&quot;</span>, n_down))</span></code></pre></div>
<pre><code>##   ↓ Downregulated in COVID-19: 277</code></pre>
</div>
<div id="volcano-plot" class="section level3">
<h3>Volcano Plot</h3>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" tabindex="-1"></a><span class="co"># Prepare volcano plot data</span></span>
<span id="cb84-2"><a href="#cb84-2" tabindex="-1"></a>volcano_data <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb84-3"><a href="#cb84-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb84-4"><a href="#cb84-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;GENEID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-5"><a href="#cb84-5" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb84-6"><a href="#cb84-6" tabindex="-1"></a>    <span class="at">sig_category =</span> <span class="fu">case_when</span>(</span>
<span id="cb84-7"><a href="#cb84-7" tabindex="-1"></a>      padj <span class="sc">&lt;</span> padj_thresh <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> lfc_thresh <span class="sc">~</span> <span class="st">&quot;Upregulated&quot;</span>,</span>
<span id="cb84-8"><a href="#cb84-8" tabindex="-1"></a>      padj <span class="sc">&lt;</span> padj_thresh <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>lfc_thresh <span class="sc">~</span> <span class="st">&quot;Downregulated&quot;</span>,</span>
<span id="cb84-9"><a href="#cb84-9" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Not Significant&quot;</span></span>
<span id="cb84-10"><a href="#cb84-10" tabindex="-1"></a>    )</span>
<span id="cb84-11"><a href="#cb84-11" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb84-12"><a href="#cb84-12" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_annot_clean, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;GENEID&quot;</span> <span class="ot">=</span> <span class="st">&quot;GeneID&quot;</span>))</span>
<span id="cb84-13"><a href="#cb84-13" tabindex="-1"></a></span>
<span id="cb84-14"><a href="#cb84-14" tabindex="-1"></a><span class="co"># Select top genes for labeling</span></span>
<span id="cb84-15"><a href="#cb84-15" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> volcano_data <span class="sc">%&gt;%</span></span>
<span id="cb84-16"><a href="#cb84-16" tabindex="-1"></a>  <span class="fu">filter</span>(sig_category <span class="sc">!=</span> <span class="st">&quot;Not Significant&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-17"><a href="#cb84-17" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb84-18"><a href="#cb84-18" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">15</span>)</span>
<span id="cb84-19"><a href="#cb84-19" tabindex="-1"></a></span>
<span id="cb84-20"><a href="#cb84-20" tabindex="-1"></a><span class="co"># Create volcano plot</span></span>
<span id="cb84-21"><a href="#cb84-21" tabindex="-1"></a><span class="fu">ggplot</span>(volcano_data, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb84-22"><a href="#cb84-22" tabindex="-1"></a>  <span class="fu">geom_point</span>(</span>
<span id="cb84-23"><a href="#cb84-23" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">color =</span> sig_category),</span>
<span id="cb84-24"><a href="#cb84-24" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb84-25"><a href="#cb84-25" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">1.5</span></span>
<span id="cb84-26"><a href="#cb84-26" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb84-27"><a href="#cb84-27" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>lfc_thresh, lfc_thresh), </span>
<span id="cb84-28"><a href="#cb84-28" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>) <span class="sc">+</span></span>
<span id="cb84-29"><a href="#cb84-29" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_thresh), </span>
<span id="cb84-30"><a href="#cb84-30" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>) <span class="sc">+</span></span>
<span id="cb84-31"><a href="#cb84-31" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb84-32"><a href="#cb84-32" tabindex="-1"></a>    <span class="at">data =</span> top_genes,</span>
<span id="cb84-33"><a href="#cb84-33" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> GENEID),</span>
<span id="cb84-34"><a href="#cb84-34" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb84-35"><a href="#cb84-35" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="dv">20</span>,</span>
<span id="cb84-36"><a href="#cb84-36" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fl">0.5</span>,</span>
<span id="cb84-37"><a href="#cb84-37" tabindex="-1"></a>    <span class="at">segment.color =</span> <span class="st">&quot;gray50&quot;</span></span>
<span id="cb84-38"><a href="#cb84-38" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb84-39"><a href="#cb84-39" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb84-40"><a href="#cb84-40" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb84-41"><a href="#cb84-41" tabindex="-1"></a>      <span class="st">&quot;Upregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;#E74C3C&quot;</span>,</span>
<span id="cb84-42"><a href="#cb84-42" tabindex="-1"></a>      <span class="st">&quot;Downregulated&quot;</span> <span class="ot">=</span> <span class="st">&quot;#3498DB&quot;</span>,</span>
<span id="cb84-43"><a href="#cb84-43" tabindex="-1"></a>      <span class="st">&quot;Not Significant&quot;</span> <span class="ot">=</span> <span class="st">&quot;gray70&quot;</span></span>
<span id="cb84-44"><a href="#cb84-44" tabindex="-1"></a>    )</span>
<span id="cb84-45"><a href="#cb84-45" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb84-46"><a href="#cb84-46" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb84-47"><a href="#cb84-47" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb84-48"><a href="#cb84-48" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb84-49"><a href="#cb84-49" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()</span>
<span id="cb84-50"><a href="#cb84-50" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb84-51"><a href="#cb84-51" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb84-52"><a href="#cb84-52" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Differential Gene Expression: COVID-19 vs. Control&quot;</span>,</span>
<span id="cb84-53"><a href="#cb84-53" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">sprintf</span>(<span class="st">&quot;%d significant DEGs (%d up, %d down)&quot;</span>, </span>
<span id="cb84-54"><a href="#cb84-54" tabindex="-1"></a>                       <span class="fu">nrow</span>(sig_degs), n_up, n_down),</span>
<span id="cb84-55"><a href="#cb84-55" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;log₂ Fold Change&quot;</span>,</span>
<span id="cb84-56"><a href="#cb84-56" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;-log_10 (Adjusted P-value)&quot;</span>,</span>
<span id="cb84-57"><a href="#cb84-57" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Classification&quot;</span></span>
<span id="cb84-58"><a href="#cb84-58" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="covid_files/figure-html/deseq_volcano-1.png" width="960" /></p>
<p>A volcano plot is a visualization <strong>used to highlight genes
that are significantly differentially expressed between our two
conditions</strong></p>
<p>The x-axis shows the direction and magnitude of the fold change,
while the y-axis shows statistical significance</p>
</div>
<div id="top-differentially-expressed-genes" class="section level3">
<h3>Top Differentially Expressed Genes</h3>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" tabindex="-1"></a><span class="co"># Format top 20 DEGs for display</span></span>
<span id="cb85-2"><a href="#cb85-2" tabindex="-1"></a>top20_degs <span class="ot">&lt;-</span> sig_degs <span class="sc">%&gt;%</span></span>
<span id="cb85-3"><a href="#cb85-3" tabindex="-1"></a>  <span class="fu">select</span>(Symbol, log2FoldChange, padj, baseMean, Description) <span class="sc">%&gt;%</span></span>
<span id="cb85-4"><a href="#cb85-4" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-5"><a href="#cb85-5" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb85-6"><a href="#cb85-6" tabindex="-1"></a>    <span class="at">log2FoldChange =</span> <span class="fu">round</span>(log2FoldChange, <span class="dv">2</span>),</span>
<span id="cb85-7"><a href="#cb85-7" tabindex="-1"></a>    <span class="at">padj =</span> <span class="fu">formatC</span>(padj, <span class="at">format =</span> <span class="st">&quot;e&quot;</span>, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb85-8"><a href="#cb85-8" tabindex="-1"></a>    <span class="at">baseMean =</span> <span class="fu">round</span>(baseMean, <span class="dv">1</span>),</span>
<span id="cb85-9"><a href="#cb85-9" tabindex="-1"></a>    <span class="at">Description =</span> <span class="fu">str_trunc</span>(Description, <span class="at">width =</span> <span class="dv">60</span>)</span>
<span id="cb85-10"><a href="#cb85-10" tabindex="-1"></a>  )</span>
<span id="cb85-11"><a href="#cb85-11" tabindex="-1"></a></span>
<span id="cb85-12"><a href="#cb85-12" tabindex="-1"></a><span class="co"># Display as formatted table</span></span>
<span id="cb85-13"><a href="#cb85-13" tabindex="-1"></a><span class="fu">kable</span>(</span>
<span id="cb85-14"><a href="#cb85-14" tabindex="-1"></a>  top20_degs,</span>
<span id="cb85-15"><a href="#cb85-15" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;log₂FC&quot;</span>, <span class="st">&quot;Adj. P-value&quot;</span>, <span class="st">&quot;Mean Expression&quot;</span>, <span class="st">&quot;Description&quot;</span>),</span>
<span id="cb85-16"><a href="#cb85-16" tabindex="-1"></a>  <span class="at">caption =</span> <span class="st">&quot;**Top 20 Differentially Expressed Genes (COVID-19 vs. Control)**&quot;</span>,</span>
<span id="cb85-17"><a href="#cb85-17" tabindex="-1"></a>  <span class="at">align =</span> <span class="fu">c</span>(<span class="st">&quot;l&quot;</span>, <span class="st">&quot;r&quot;</span>, <span class="st">&quot;r&quot;</span>, <span class="st">&quot;r&quot;</span>, <span class="st">&quot;l&quot;</span>)</span>
<span id="cb85-18"><a href="#cb85-18" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb85-19"><a href="#cb85-19" tabindex="-1"></a>  <span class="fu">kable_styling</span>(</span>
<span id="cb85-20"><a href="#cb85-20" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>),</span>
<span id="cb85-21"><a href="#cb85-21" tabindex="-1"></a>    <span class="at">full_width =</span> <span class="cn">FALSE</span></span>
<span id="cb85-22"><a href="#cb85-22" tabindex="-1"></a>  )</span></code></pre></div>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<strong>Top 20 Differentially Expressed Genes (COVID-19
vs. Control)</strong>
</caption>
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
log₂FC
</th>
<th style="text-align:right;">
Adj. P-value
</th>
<th style="text-align:right;">
Mean Expression
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IRAK2
</td>
<td style="text-align:right;">
1.81
</td>
<td style="text-align:right;">
4.50e-19
</td>
<td style="text-align:right;">
704.8
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:right;">
1.45
</td>
<td style="text-align:right;">
4.65e-19
</td>
<td style="text-align:right;">
1800.8
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
BIRC3
</td>
<td style="text-align:right;">
1.69
</td>
<td style="text-align:right;">
2.44e-18
</td>
<td style="text-align:right;">
1057.1
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
TNFAIP3
</td>
<td style="text-align:right;">
3.08
</td>
<td style="text-align:right;">
2.93e-16
</td>
<td style="text-align:right;">
2141.0
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
BCL2A1
</td>
<td style="text-align:right;">
2.35
</td>
<td style="text-align:right;">
2.08e-15
</td>
<td style="text-align:right;">
98.5
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
IL1A
</td>
<td style="text-align:right;">
4.02
</td>
<td style="text-align:right;">
3.16e-15
</td>
<td style="text-align:right;">
873.1
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
STAT1
</td>
<td style="text-align:right;">
1.24
</td>
<td style="text-align:right;">
6.36e-15
</td>
<td style="text-align:right;">
2879.2
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8L1
</td>
<td style="text-align:right;">
-25.37
</td>
<td style="text-align:right;">
2.83e-13
</td>
<td style="text-align:right;">
58.4
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
RASGRP1
</td>
<td style="text-align:right;">
1.47
</td>
<td style="text-align:right;">
3.21e-13
</td>
<td style="text-align:right;">
60.7
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
PARP9
</td>
<td style="text-align:right;">
1.74
</td>
<td style="text-align:right;">
1.50e-12
</td>
<td style="text-align:right;">
574.4
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8
</td>
<td style="text-align:right;">
-24.51
</td>
<td style="text-align:right;">
2.11e-12
</td>
<td style="text-align:right;">
37.4
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
DDX60
</td>
<td style="text-align:right;">
2.34
</td>
<td style="text-align:right;">
4.47e-12
</td>
<td style="text-align:right;">
319.7
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
IL36G
</td>
<td style="text-align:right;">
2.24
</td>
<td style="text-align:right;">
1.50e-11
</td>
<td style="text-align:right;">
32.8
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
MAFF
</td>
<td style="text-align:right;">
1.98
</td>
<td style="text-align:right;">
1.81e-11
</td>
<td style="text-align:right;">
527.5
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
TIPARP
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
1.97e-11
</td>
<td style="text-align:right;">
1792.9
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ZC3H12C
</td>
<td style="text-align:right;">
1.39
</td>
<td style="text-align:right;">
5.04e-11
</td>
<td style="text-align:right;">
217.5
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
DTX3L
</td>
<td style="text-align:right;">
1.40
</td>
<td style="text-align:right;">
5.04e-11
</td>
<td style="text-align:right;">
598.0
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NFKBIA
</td>
<td style="text-align:right;">
2.28
</td>
<td style="text-align:right;">
1.00e-10
</td>
<td style="text-align:right;">
3803.9
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SAMD4A
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
1.00e-10
</td>
<td style="text-align:right;">
297.2
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
HBEGF
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
1.00e-10
</td>
<td style="text-align:right;">
201.5
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<hr />
</div>
</div>
<div id="targeted-gene-analysis" class="section level2">
<h2>Targeted Gene Analysis</h2>
<div id="select-genes-of-interest" class="section level3">
<h3>Select Genes of Interest</h3>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" tabindex="-1"></a><span class="co"># Define your 4 genes of interest by symbol</span></span>
<span id="cb86-2"><a href="#cb86-2" tabindex="-1"></a>genes_of_interest_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;IRAK2&quot;</span>, <span class="st">&quot;SOD2&quot;</span>, <span class="st">&quot;ANXA8L1&quot;</span>, <span class="st">&quot;ANXA8&quot;</span>)</span>
<span id="cb86-3"><a href="#cb86-3" tabindex="-1"></a></span>
<span id="cb86-4"><a href="#cb86-4" tabindex="-1"></a><span class="co"># Find corresponding Ensembl IDs</span></span>
<span id="cb86-5"><a href="#cb86-5" tabindex="-1"></a>target_genes <span class="ot">&lt;-</span> sig_degs <span class="sc">%&gt;%</span></span>
<span id="cb86-6"><a href="#cb86-6" tabindex="-1"></a>  <span class="fu">filter</span>(Symbol <span class="sc">%in%</span> genes_of_interest_symbols) <span class="sc">%&gt;%</span></span>
<span id="cb86-7"><a href="#cb86-7" tabindex="-1"></a>  <span class="fu">select</span>(GENEID, Symbol, log2FoldChange, padj, baseMean, Description)</span>
<span id="cb86-8"><a href="#cb86-8" tabindex="-1"></a></span>
<span id="cb86-9"><a href="#cb86-9" tabindex="-1"></a><span class="co"># Check if all genes were found</span></span>
<span id="cb86-10"><a href="#cb86-10" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(target_genes) <span class="sc">&lt;</span> <span class="fu">length</span>(genes_of_interest_symbols)) {</span>
<span id="cb86-11"><a href="#cb86-11" tabindex="-1"></a>  missing <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(genes_of_interest_symbols, target_genes<span class="sc">$</span>Symbol)</span>
<span id="cb86-12"><a href="#cb86-12" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;⚠ Warning: The following genes were not found in DEG results:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb86-13"><a href="#cb86-13" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste</span>(<span class="st">&quot;  -&quot;</span>, missing, <span class="at">collapse =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb86-14"><a href="#cb86-14" tabindex="-1"></a>  </span>
<span id="cb86-15"><a href="#cb86-15" tabindex="-1"></a>  <span class="co"># Search in full annotation to get their IDs</span></span>
<span id="cb86-16"><a href="#cb86-16" tabindex="-1"></a>  additional_genes <span class="ot">&lt;-</span> gene_annot_clean <span class="sc">%&gt;%</span></span>
<span id="cb86-17"><a href="#cb86-17" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">%in%</span> missing) <span class="sc">%&gt;%</span></span>
<span id="cb86-18"><a href="#cb86-18" tabindex="-1"></a>    <span class="fu">left_join</span>(</span>
<span id="cb86-19"><a href="#cb86-19" tabindex="-1"></a>      res <span class="sc">%&gt;%</span> </span>
<span id="cb86-20"><a href="#cb86-20" tabindex="-1"></a>        <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb86-21"><a href="#cb86-21" tabindex="-1"></a>        <span class="fu">rownames_to_column</span>(<span class="st">&quot;GENEID&quot;</span>),</span>
<span id="cb86-22"><a href="#cb86-22" tabindex="-1"></a>      <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span> <span class="ot">=</span> <span class="st">&quot;GENEID&quot;</span>)</span>
<span id="cb86-23"><a href="#cb86-23" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb86-24"><a href="#cb86-24" tabindex="-1"></a>    <span class="fu">select</span>(GeneID, Symbol, log2FoldChange, padj, baseMean, Description) <span class="sc">%&gt;%</span></span>
<span id="cb86-25"><a href="#cb86-25" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">GENEID =</span> GeneID)</span>
<span id="cb86-26"><a href="#cb86-26" tabindex="-1"></a>  </span>
<span id="cb86-27"><a href="#cb86-27" tabindex="-1"></a>  target_genes <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(target_genes, additional_genes)</span>
<span id="cb86-28"><a href="#cb86-28" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;✓ Found %d/%d genes in full dataset</span><span class="sc">\n\n</span><span class="st">&quot;</span>, </span>
<span id="cb86-29"><a href="#cb86-29" tabindex="-1"></a>              <span class="fu">nrow</span>(target_genes), </span>
<span id="cb86-30"><a href="#cb86-30" tabindex="-1"></a>              <span class="fu">length</span>(genes_of_interest_symbols)))</span>
<span id="cb86-31"><a href="#cb86-31" tabindex="-1"></a>}</span>
<span id="cb86-32"><a href="#cb86-32" tabindex="-1"></a></span>
<span id="cb86-33"><a href="#cb86-33" tabindex="-1"></a><span class="co"># Get the Ensembl IDs for downstream use</span></span>
<span id="cb86-34"><a href="#cb86-34" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> target_genes<span class="sc">$</span>GENEID</span>
<span id="cb86-35"><a href="#cb86-35" tabindex="-1"></a></span>
<span id="cb86-36"><a href="#cb86-36" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;**Genes Selected for Analysis:**</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## **Genes Selected for Analysis:**</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" tabindex="-1"></a><span class="fu">kable</span>(target_genes, </span>
<span id="cb88-2"><a href="#cb88-2" tabindex="-1"></a>      <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Gene ID&quot;</span>, <span class="st">&quot;Symbol&quot;</span>, <span class="st">&quot;log₂FC&quot;</span>, <span class="st">&quot;Adj. P-value&quot;</span>, </span>
<span id="cb88-3"><a href="#cb88-3" tabindex="-1"></a>                    <span class="st">&quot;Mean Expr.&quot;</span>, <span class="st">&quot;Description&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb88-4"><a href="#cb88-4" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>))</span></code></pre></div>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene ID
</th>
<th style="text-align:left;">
Symbol
</th>
<th style="text-align:right;">
log₂FC
</th>
<th style="text-align:right;">
Adj. P-value
</th>
<th style="text-align:right;">
Mean Expr.
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IRAK2
</td>
<td style="text-align:left;">
IRAK2
</td>
<td style="text-align:right;">
1.805490
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
704.75879
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:right;">
1.451888
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1800.75591
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8L1
</td>
<td style="text-align:left;">
ANXA8L1
</td>
<td style="text-align:right;">
-25.366651
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
58.42197
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8
</td>
<td style="text-align:left;">
ANXA8
</td>
<td style="text-align:right;">
-24.513486
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
37.43467
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<div id="expression-patterns-across-conditions" class="section level3">
<h3>Expression Patterns Across Conditions</h3>
<pre><code>## ⚠ Error: None of the genes found in expression matrix.
## Checking if genes need different ID format...
## 
## ✓ Found 4 genes using alternative matching</code></pre>
<pre><code>## Genes available in expression matrix: 4/4</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" tabindex="-1"></a><span class="co"># Extract expression data for available genes</span></span>
<span id="cb91-2"><a href="#cb91-2" tabindex="-1"></a>gene_expr_data <span class="ot">&lt;-</span> log_mat[available_genes, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">%&gt;%</span></span>
<span id="cb91-3"><a href="#cb91-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb91-4"><a href="#cb91-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;GENEID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-5"><a href="#cb91-5" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb91-6"><a href="#cb91-6" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>GENEID,</span>
<span id="cb91-7"><a href="#cb91-7" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb91-8"><a href="#cb91-8" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;expression&quot;</span></span>
<span id="cb91-9"><a href="#cb91-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb91-10"><a href="#cb91-10" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_annot, <span class="at">by =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-11"><a href="#cb91-11" tabindex="-1"></a>  <span class="fu">left_join</span>(target_genes <span class="sc">%&gt;%</span> <span class="fu">select</span>(GENEID, Symbol), <span class="at">by =</span> <span class="st">&quot;GENEID&quot;</span>)</span>
<span id="cb91-12"><a href="#cb91-12" tabindex="-1"></a></span>
<span id="cb91-13"><a href="#cb91-13" tabindex="-1"></a>gene_expr_data <span class="ot">&lt;-</span> gene_expr_data <span class="sc">%&gt;%</span></span>
<span id="cb91-14"><a href="#cb91-14" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb91-15"><a href="#cb91-15" tabindex="-1"></a>    <span class="at">Name =</span> <span class="fu">recode</span>(</span>
<span id="cb91-16"><a href="#cb91-16" tabindex="-1"></a>      GENEID,</span>
<span id="cb91-17"><a href="#cb91-17" tabindex="-1"></a>      <span class="st">&quot;3656&quot;</span> <span class="ot">=</span> <span class="st">&quot;IRAK2&quot;</span>,</span>
<span id="cb91-18"><a href="#cb91-18" tabindex="-1"></a>      <span class="st">&quot;6648&quot;</span> <span class="ot">=</span> <span class="st">&quot;SOD2&quot;</span>,</span>
<span id="cb91-19"><a href="#cb91-19" tabindex="-1"></a>      <span class="st">&quot;728113&quot;</span> <span class="ot">=</span> <span class="st">&quot;ANXA8L1&quot;</span>,</span>
<span id="cb91-20"><a href="#cb91-20" tabindex="-1"></a>      <span class="st">&quot;653145&quot;</span> <span class="ot">=</span> <span class="st">&quot;ANXA8&quot;</span></span>
<span id="cb91-21"><a href="#cb91-21" tabindex="-1"></a>    )</span>
<span id="cb91-22"><a href="#cb91-22" tabindex="-1"></a>  )</span>
<span id="cb91-23"><a href="#cb91-23" tabindex="-1"></a></span>
<span id="cb91-24"><a href="#cb91-24" tabindex="-1"></a><span class="co"># Boxplot by condition</span></span>
<span id="cb91-25"><a href="#cb91-25" tabindex="-1"></a><span class="fu">ggplot</span>(gene_expr_data, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> expression, <span class="at">fill =</span> group)) <span class="sc">+</span></span>
<span id="cb91-26"><a href="#cb91-26" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb91-27"><a href="#cb91-27" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb91-28"><a href="#cb91-28" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Name, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb91-29"><a href="#cb91-29" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb91-30"><a href="#cb91-30" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb91-31"><a href="#cb91-31" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb91-32"><a href="#cb91-32" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb91-33"><a href="#cb91-33" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb91-34"><a href="#cb91-34" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb91-35"><a href="#cb91-35" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Expression of Selected Genes Across Conditions&quot;</span>,</span>
<span id="cb91-36"><a href="#cb91-36" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb91-37"><a href="#cb91-37" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;log₂(FPKM + 1)&quot;</span></span>
<span id="cb91-38"><a href="#cb91-38" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_expression_plot-1.png" width="960" /></p>
<p><strong>This boxplot shows the distribution of expression values for
each selected gene across all experimental conditions</strong>.</p>
<p>The plot highlights both upregulation and downregulation patterns of
the genes across conditions, while also revealing variability among
replicates.</p>
</div>
<div id="covid-19-vs-control-comparison" class="section level3">
<h3>COVID-19 vs Control Comparison</h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" tabindex="-1"></a><span class="co"># Focus on COVID-19 vs Control comparison</span></span>
<span id="cb92-2"><a href="#cb92-2" tabindex="-1"></a>covid_comparison <span class="ot">&lt;-</span> gene_expr_data <span class="sc">%&gt;%</span></span>
<span id="cb92-3"><a href="#cb92-3" tabindex="-1"></a>  <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;COVID19&quot;</span>, <span class="st">&quot;Healthy&quot;</span>, <span class="st">&quot;Mock&quot;</span>, <span class="st">&quot;SARS2&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-4"><a href="#cb92-4" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb92-5"><a href="#cb92-5" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">ifelse</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;COVID19&quot;</span>, <span class="st">&quot;SARS2&quot;</span>), <span class="st">&quot;Infected&quot;</span>, <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb92-6"><a href="#cb92-6" tabindex="-1"></a>  )</span>
<span id="cb92-7"><a href="#cb92-7" tabindex="-1"></a></span>
<span id="cb92-8"><a href="#cb92-8" tabindex="-1"></a><span class="co"># Bar plot with error bars</span></span>
<span id="cb92-9"><a href="#cb92-9" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> covid_comparison <span class="sc">%&gt;%</span></span>
<span id="cb92-10"><a href="#cb92-10" tabindex="-1"></a>  <span class="fu">group_by</span>(Name, condition) <span class="sc">%&gt;%</span></span>
<span id="cb92-11"><a href="#cb92-11" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb92-12"><a href="#cb92-12" tabindex="-1"></a>    <span class="at">mean_expr =</span> <span class="fu">mean</span>(expression, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb92-13"><a href="#cb92-13" tabindex="-1"></a>    <span class="at">se_expr =</span> <span class="fu">sd</span>(expression, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">n</span>()),</span>
<span id="cb92-14"><a href="#cb92-14" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span></span>
<span id="cb92-15"><a href="#cb92-15" tabindex="-1"></a>  )</span>
<span id="cb92-16"><a href="#cb92-16" tabindex="-1"></a></span>
<span id="cb92-17"><a href="#cb92-17" tabindex="-1"></a><span class="fu">ggplot</span>(summary_stats, <span class="fu">aes</span>(<span class="at">x =</span> Name, <span class="at">y =</span> mean_expr, <span class="at">fill =</span> condition)) <span class="sc">+</span></span>
<span id="cb92-18"><a href="#cb92-18" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb92-19"><a href="#cb92-19" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(</span>
<span id="cb92-20"><a href="#cb92-20" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">ymin =</span> mean_expr <span class="sc">-</span> se_expr, <span class="at">ymax =</span> mean_expr <span class="sc">+</span> se_expr),</span>
<span id="cb92-21"><a href="#cb92-21" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),</span>
<span id="cb92-22"><a href="#cb92-22" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.25</span></span>
<span id="cb92-23"><a href="#cb92-23" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb92-24"><a href="#cb92-24" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span> <span class="ot">=</span> <span class="st">&quot;#3498DB&quot;</span>, <span class="st">&quot;Infected&quot;</span> <span class="ot">=</span> <span class="st">&quot;#E74C3C&quot;</span>)) <span class="sc">+</span></span>
<span id="cb92-25"><a href="#cb92-25" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb92-26"><a href="#cb92-26" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb92-27"><a href="#cb92-27" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb92-28"><a href="#cb92-28" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Gene Expression: COVID-19/SARS-CoV-2 vs. Controls&quot;</span>,</span>
<span id="cb92-29"><a href="#cb92-29" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">&quot;Mean ± SE&quot;</span>,</span>
<span id="cb92-30"><a href="#cb92-30" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb92-31"><a href="#cb92-31" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;log₂(FPKM + 1)&quot;</span>,</span>
<span id="cb92-32"><a href="#cb92-32" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;Condition&quot;</span></span>
<span id="cb92-33"><a href="#cb92-33" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_comparison-1.png" width="960" /></p>
<p>This bar plot compares the expression of selected genes between
COVID-19 infected samples and controls. <strong>Each bar represents the
mean expression of a gene, with error bars showing standard error across
samples</strong></p>
</div>
<div id="correlation-between-genes" class="section level3">
<h3>Correlation Between Genes</h3>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" tabindex="-1"></a><span class="co"># Define the exact gene symbols you want</span></span>
<span id="cb93-2"><a href="#cb93-2" tabindex="-1"></a>genes_of_interest_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;IRAK2&quot;</span>, <span class="st">&quot;SOD2&quot;</span>, <span class="st">&quot;ANXA8L1&quot;</span>, <span class="st">&quot;ANXA8&quot;</span>)</span>
<span id="cb93-3"><a href="#cb93-3" tabindex="-1"></a></span>
<span id="cb93-4"><a href="#cb93-4" tabindex="-1"></a><span class="co"># Find corresponding Ensembl IDs from gene annotation</span></span>
<span id="cb93-5"><a href="#cb93-5" tabindex="-1"></a>target_genes_fixed <span class="ot">&lt;-</span> gene_annot_clean <span class="sc">%&gt;%</span></span>
<span id="cb93-6"><a href="#cb93-6" tabindex="-1"></a>  <span class="fu">filter</span>(Symbol <span class="sc">%in%</span> genes_of_interest_symbols)</span>
<span id="cb93-7"><a href="#cb93-7" tabindex="-1"></a></span>
<span id="cb93-8"><a href="#cb93-8" tabindex="-1"></a><span class="co"># Get the gene IDs that are available in expression matrix</span></span>
<span id="cb93-9"><a href="#cb93-9" tabindex="-1"></a>available_genes <span class="ot">&lt;-</span> target_genes_fixed<span class="sc">$</span>GeneID[target_genes_fixed<span class="sc">$</span>GeneID <span class="sc">%in%</span> <span class="fu">rownames</span>(log_mat)]</span>
<span id="cb93-10"><a href="#cb93-10" tabindex="-1"></a></span>
<span id="cb93-11"><a href="#cb93-11" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Available genes found:&quot;</span>, <span class="fu">length</span>(available_genes), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Available genes found: 4</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Gene IDs:&quot;</span>, available_genes, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Gene IDs: 3656 6648 728113 653145</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" tabindex="-1"></a><span class="co"># Extract expression data</span></span>
<span id="cb97-2"><a href="#cb97-2" tabindex="-1"></a>gene_expr_matrix <span class="ot">&lt;-</span> log_mat[available_genes, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb97-3"><a href="#cb97-3" tabindex="-1"></a></span>
<span id="cb97-4"><a href="#cb97-4" tabindex="-1"></a><span class="co"># Create proper mapping: GeneID -&gt; Symbol</span></span>
<span id="cb97-5"><a href="#cb97-5" tabindex="-1"></a>gene_id_to_symbol <span class="ot">&lt;-</span> target_genes_fixed <span class="sc">%&gt;%</span></span>
<span id="cb97-6"><a href="#cb97-6" tabindex="-1"></a>  <span class="fu">filter</span>(GeneID <span class="sc">%in%</span> available_genes) <span class="sc">%&gt;%</span></span>
<span id="cb97-7"><a href="#cb97-7" tabindex="-1"></a>  <span class="fu">select</span>(GeneID, Symbol)</span>
<span id="cb97-8"><a href="#cb97-8" tabindex="-1"></a></span>
<span id="cb97-9"><a href="#cb97-9" tabindex="-1"></a><span class="co"># Map the rownames to symbols</span></span>
<span id="cb97-10"><a href="#cb97-10" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> gene_id_to_symbol<span class="sc">$</span>Symbol[<span class="fu">match</span>(available_genes, gene_id_to_symbol<span class="sc">$</span>GeneID)]</span>
<span id="cb97-11"><a href="#cb97-11" tabindex="-1"></a></span>
<span id="cb97-12"><a href="#cb97-12" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Gene symbols:&quot;</span>, gene_symbols, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## Gene symbols: IRAK2 SOD2 ANXA8L1 ANXA8</code></pre>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" tabindex="-1"></a><span class="co"># Assign symbols as rownames</span></span>
<span id="cb99-2"><a href="#cb99-2" tabindex="-1"></a><span class="fu">rownames</span>(gene_expr_matrix) <span class="ot">&lt;-</span> gene_symbols</span>
<span id="cb99-3"><a href="#cb99-3" tabindex="-1"></a></span>
<span id="cb99-4"><a href="#cb99-4" tabindex="-1"></a><span class="co"># Correlation matrix</span></span>
<span id="cb99-5"><a href="#cb99-5" tabindex="-1"></a>gene_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(<span class="fu">t</span>(gene_expr_matrix), <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>)</span>
<span id="cb99-6"><a href="#cb99-6" tabindex="-1"></a></span>
<span id="cb99-7"><a href="#cb99-7" tabindex="-1"></a><span class="co"># Set dimension names</span></span>
<span id="cb99-8"><a href="#cb99-8" tabindex="-1"></a><span class="fu">dimnames</span>(gene_cor) <span class="ot">&lt;-</span> <span class="fu">list</span>(gene_symbols, gene_symbols)</span>
<span id="cb99-9"><a href="#cb99-9" tabindex="-1"></a></span>
<span id="cb99-10"><a href="#cb99-10" tabindex="-1"></a><span class="co"># Correlation heatmap</span></span>
<span id="cb99-11"><a href="#cb99-11" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb99-12"><a href="#cb99-12" tabindex="-1"></a>  gene_cor,</span>
<span id="cb99-13"><a href="#cb99-13" tabindex="-1"></a>  <span class="at">display_numbers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-14"><a href="#cb99-14" tabindex="-1"></a>  <span class="at">number_format =</span> <span class="st">&quot;%.2f&quot;</span>,</span>
<span id="cb99-15"><a href="#cb99-15" tabindex="-1"></a>  <span class="at">number_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb99-16"><a href="#cb99-16" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;#3498DB&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#E74C3C&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb99-17"><a href="#cb99-17" tabindex="-1"></a>  <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb99-18"><a href="#cb99-18" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Correlation Between Selected Genes&quot;</span>,</span>
<span id="cb99-19"><a href="#cb99-19" tabindex="-1"></a>  <span class="at">fontsize =</span> <span class="dv">12</span>,</span>
<span id="cb99-20"><a href="#cb99-20" tabindex="-1"></a>  <span class="at">fontsize_number =</span> <span class="dv">10</span>,</span>
<span id="cb99-21"><a href="#cb99-21" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-22"><a href="#cb99-22" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span></span>
<span id="cb99-23"><a href="#cb99-23" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_correlation-1.png" width="960" /></p>
<p>A correlation plot shows how strongly the expression of selected
genes changes together across all samples</p>
<p>How to interpret - A value close to +1 = genes increase together
(positive correlation). - A value close to –1 = genes move in opposite
directions. - A value near 0 = no relationship.</p>
<p><strong>This correlation plot shows how the four genes we chose
relate to each other</strong>. ANXA8 and ANXA8L1 tightly track together,
both being heavily downregulated, while IRAK2 and SOD2 rise together in
COVID-19 samples. These coordinated patterns reinforce that they are
involved in shared pathways</p>
<div class="sourceCode" id="cb100"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" tabindex="-1"></a><span class="co"># Pairwise scatterplots if we have 4 genes</span></span>
<span id="cb100-2"><a href="#cb100-2" tabindex="-1"></a><span class="co"># if (nrow(gene_expr_matrix) &gt;= 2) {</span></span>
<span id="cb100-3"><a href="#cb100-3" tabindex="-1"></a><span class="co">#   gene_symbols &lt;- rownames(gene_expr_matrix)</span></span>
<span id="cb100-4"><a href="#cb100-4" tabindex="-1"></a><span class="co">#   n_genes &lt;- length(gene_symbols)</span></span>
<span id="cb100-5"><a href="#cb100-5" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb100-6"><a href="#cb100-6" tabindex="-1"></a><span class="co">#   if (n_genes == 4) {</span></span>
<span id="cb100-7"><a href="#cb100-7" tabindex="-1"></a><span class="co">#     par(mfrow = c(2, 3), mar = c(4, 4, 2, 1))</span></span>
<span id="cb100-8"><a href="#cb100-8" tabindex="-1"></a><span class="co">#     </span></span>
<span id="cb100-9"><a href="#cb100-9" tabindex="-1"></a><span class="co">#     # All pairwise combinations</span></span>
<span id="cb100-10"><a href="#cb100-10" tabindex="-1"></a><span class="co">#     pairs_to_plot &lt;- combn(1:4, 2)</span></span>
<span id="cb100-11"><a href="#cb100-11" tabindex="-1"></a><span class="co">#     </span></span>
<span id="cb100-12"><a href="#cb100-12" tabindex="-1"></a><span class="co">#     for (i in 1:ncol(pairs_to_plot)) {</span></span>
<span id="cb100-13"><a href="#cb100-13" tabindex="-1"></a><span class="co">#       idx1 &lt;- pairs_to_plot[1, i]</span></span>
<span id="cb100-14"><a href="#cb100-14" tabindex="-1"></a><span class="co">#       idx2 &lt;- pairs_to_plot[2, i]</span></span>
<span id="cb100-15"><a href="#cb100-15" tabindex="-1"></a><span class="co">#       </span></span>
<span id="cb100-16"><a href="#cb100-16" tabindex="-1"></a><span class="co">#       x &lt;- gene_expr_matrix[idx1, ]</span></span>
<span id="cb100-17"><a href="#cb100-17" tabindex="-1"></a><span class="co">#       y &lt;- gene_expr_matrix[idx2, ]</span></span>
<span id="cb100-18"><a href="#cb100-18" tabindex="-1"></a><span class="co">#       </span></span>
<span id="cb100-19"><a href="#cb100-19" tabindex="-1"></a><span class="co">#       plot(x, y,</span></span>
<span id="cb100-20"><a href="#cb100-20" tabindex="-1"></a><span class="co">#            xlab = gene_symbols[idx1],</span></span>
<span id="cb100-21"><a href="#cb100-21" tabindex="-1"></a><span class="co">#            ylab = gene_symbols[idx2],</span></span>
<span id="cb100-22"><a href="#cb100-22" tabindex="-1"></a><span class="co">#            main = sprintf(&quot;r = %.2f&quot;, cor(x, y)),</span></span>
<span id="cb100-23"><a href="#cb100-23" tabindex="-1"></a><span class="co">#            pch = 19,</span></span>
<span id="cb100-24"><a href="#cb100-24" tabindex="-1"></a><span class="co">#            col = rgb(0, 0, 0, 0.5))</span></span>
<span id="cb100-25"><a href="#cb100-25" tabindex="-1"></a><span class="co">#       abline(lm(y ~ x), col = &quot;red&quot;, lwd = 2)</span></span>
<span id="cb100-26"><a href="#cb100-26" tabindex="-1"></a><span class="co">#     }</span></span>
<span id="cb100-27"><a href="#cb100-27" tabindex="-1"></a><span class="co">#     </span></span>
<span id="cb100-28"><a href="#cb100-28" tabindex="-1"></a><span class="co">#     par(mfrow = c(1, 1))</span></span>
<span id="cb100-29"><a href="#cb100-29" tabindex="-1"></a><span class="co">#   } else if (n_genes &gt; 1) {</span></span>
<span id="cb100-30"><a href="#cb100-30" tabindex="-1"></a><span class="co">#     # For other numbers of genes, use pairs() function</span></span>
<span id="cb100-31"><a href="#cb100-31" tabindex="-1"></a><span class="co">#     pairs(t(gene_expr_matrix), </span></span>
<span id="cb100-32"><a href="#cb100-32" tabindex="-1"></a><span class="co">#           main = &quot;Pairwise Gene Expression Correlations&quot;,</span></span>
<span id="cb100-33"><a href="#cb100-33" tabindex="-1"></a><span class="co">#           pch = 19,</span></span>
<span id="cb100-34"><a href="#cb100-34" tabindex="-1"></a><span class="co">#           col = rgb(0, 0, 0, 0.5))</span></span>
<span id="cb100-35"><a href="#cb100-35" tabindex="-1"></a><span class="co">#   }</span></span>
<span id="cb100-36"><a href="#cb100-36" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb100-37"><a href="#cb100-37" tabindex="-1"></a></span>
<span id="cb100-38"><a href="#cb100-38" tabindex="-1"></a><span class="co"># Prepare the matrix: samples as rows, genes as columns</span></span>
<span id="cb100-39"><a href="#cb100-39" tabindex="-1"></a>gene_matrix <span class="ot">&lt;-</span> <span class="fu">t</span>(gene_expr_matrix)  <span class="co"># genes x samples → samples x genes</span></span>
<span id="cb100-40"><a href="#cb100-40" tabindex="-1"></a><span class="fu">colnames</span>(gene_matrix) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gene_expr_matrix)  <span class="co"># set gene names</span></span>
<span id="cb100-41"><a href="#cb100-41" tabindex="-1"></a></span>
<span id="cb100-42"><a href="#cb100-42" tabindex="-1"></a><span class="co"># Custom panel function to add points + regression line</span></span>
<span id="cb100-43"><a href="#cb100-43" tabindex="-1"></a>panel_regression <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb100-44"><a href="#cb100-44" tabindex="-1"></a>  <span class="fu">points</span>(x, y, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="fu">rgb</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.5</span>))</span>
<span id="cb100-45"><a href="#cb100-45" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="fu">lm</span>(y <span class="sc">~</span> x), <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)  <span class="co"># linear regression line</span></span>
<span id="cb100-46"><a href="#cb100-46" tabindex="-1"></a>}</span>
<span id="cb100-47"><a href="#cb100-47" tabindex="-1"></a></span>
<span id="cb100-48"><a href="#cb100-48" tabindex="-1"></a><span class="co"># Draw labeled scatterplot matrix</span></span>
<span id="cb100-49"><a href="#cb100-49" tabindex="-1"></a><span class="fu">pairs</span>(</span>
<span id="cb100-50"><a href="#cb100-50" tabindex="-1"></a>  gene_matrix,</span>
<span id="cb100-51"><a href="#cb100-51" tabindex="-1"></a>  <span class="at">upper.panel =</span> panel_regression,  <span class="co"># show points + regression in upper panels</span></span>
<span id="cb100-52"><a href="#cb100-52" tabindex="-1"></a>  <span class="at">lower.panel =</span> panel_regression,  <span class="co"># optionally same for lower</span></span>
<span id="cb100-53"><a href="#cb100-53" tabindex="-1"></a>  <span class="at">diag.panel =</span> <span class="cf">function</span>(x) {       <span class="co"># histogram on diagonal</span></span>
<span id="cb100-54"><a href="#cb100-54" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-55"><a href="#cb100-55" tabindex="-1"></a>    <span class="co">#hist(x, col = &quot;lightgray&quot;, main = &quot;&quot;, axes = FALSE)</span></span>
<span id="cb100-56"><a href="#cb100-56" tabindex="-1"></a>  },</span>
<span id="cb100-57"><a href="#cb100-57" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Pairwise Gene Expression Correlations&quot;</span></span>
<span id="cb100-58"><a href="#cb100-58" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_correlation2-1.png" width="960" /></p>
<p>These pairwise scatterplots show how the expression of the four
selected genes correlates across all samples. <strong>Each plot shows
one gene on the x-axis and another on the y-axis</strong>, with a linear
regression line overlaid in red.</p>
</div>
<div id="heatmap-of-gene-expression" class="section level3">
<h3>Heatmap of Gene Expression</h3>
<div class="sourceCode" id="cb101"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" tabindex="-1"></a><span class="co"># Prepare annotation for samples</span></span>
<span id="cb101-2"><a href="#cb101-2" tabindex="-1"></a>annotation_col <span class="ot">&lt;-</span> sample_annot <span class="sc">%&gt;%</span></span>
<span id="cb101-3"><a href="#cb101-3" tabindex="-1"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> <span class="fu">colnames</span>(gene_expr_matrix)) <span class="sc">%&gt;%</span></span>
<span id="cb101-4"><a href="#cb101-4" tabindex="-1"></a>  <span class="fu">select</span>(sample, group, cell_type) <span class="sc">%&gt;%</span></span>
<span id="cb101-5"><a href="#cb101-5" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb101-6"><a href="#cb101-6" tabindex="-1"></a></span>
<span id="cb101-7"><a href="#cb101-7" tabindex="-1"></a><span class="co"># Scale expression for visualization</span></span>
<span id="cb101-8"><a href="#cb101-8" tabindex="-1"></a>gene_expr_scaled <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(gene_expr_matrix)))</span>
<span id="cb101-9"><a href="#cb101-9" tabindex="-1"></a></span>
<span id="cb101-10"><a href="#cb101-10" tabindex="-1"></a><span class="co"># Heatmap</span></span>
<span id="cb101-11"><a href="#cb101-11" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb101-12"><a href="#cb101-12" tabindex="-1"></a>  gene_expr_scaled,</span>
<span id="cb101-13"><a href="#cb101-13" tabindex="-1"></a>  <span class="at">annotation_col =</span> annotation_col,</span>
<span id="cb101-14"><a href="#cb101-14" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb101-15"><a href="#cb101-15" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb101-16"><a href="#cb101-16" tabindex="-1"></a>  <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb101-17"><a href="#cb101-17" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;#3498DB&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#E74C3C&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb101-18"><a href="#cb101-18" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Expression Heatmap of Selected Genes</span><span class="sc">\n</span><span class="st">(Z-score normalized)&quot;</span>,</span>
<span id="cb101-19"><a href="#cb101-19" tabindex="-1"></a>  <span class="at">fontsize_row =</span> <span class="dv">10</span></span>
<span id="cb101-20"><a href="#cb101-20" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_heatmap-1.png" width="960" /></p>
<p>A heatmap displays the expression levels of selected genes across all
samples, with colors representing high or low expression.</p>
</div>
<div id="statistical-testing" class="section level3">
<h3>Statistical Testing</h3>
<div class="sourceCode" id="cb102"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" tabindex="-1"></a><span class="co"># Perform t-tests for each gene (COVID/SARS2 vs Control)</span></span>
<span id="cb102-2"><a href="#cb102-2" tabindex="-1"></a>covid_samples <span class="ot">&lt;-</span> sample_annot <span class="sc">%&gt;%</span></span>
<span id="cb102-3"><a href="#cb102-3" tabindex="-1"></a>  <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;COVID19&quot;</span>, <span class="st">&quot;SARS2&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb102-4"><a href="#cb102-4" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb102-5"><a href="#cb102-5" tabindex="-1"></a></span>
<span id="cb102-6"><a href="#cb102-6" tabindex="-1"></a>control_samples <span class="ot">&lt;-</span> sample_annot <span class="sc">%&gt;%</span></span>
<span id="cb102-7"><a href="#cb102-7" tabindex="-1"></a>  <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Healthy&quot;</span>, <span class="st">&quot;Mock&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb102-8"><a href="#cb102-8" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb102-9"><a href="#cb102-9" tabindex="-1"></a></span>
<span id="cb102-10"><a href="#cb102-10" tabindex="-1"></a><span class="co"># Filter to available samples in expression matrix</span></span>
<span id="cb102-11"><a href="#cb102-11" tabindex="-1"></a>covid_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(covid_samples, <span class="fu">colnames</span>(gene_expr_matrix))</span>
<span id="cb102-12"><a href="#cb102-12" tabindex="-1"></a>control_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(control_samples, <span class="fu">colnames</span>(gene_expr_matrix))</span>
<span id="cb102-13"><a href="#cb102-13" tabindex="-1"></a></span>
<span id="cb102-14"><a href="#cb102-14" tabindex="-1"></a><span class="co"># T-test for each gene</span></span>
<span id="cb102-15"><a href="#cb102-15" tabindex="-1"></a>test_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(gene_expr_matrix), <span class="cf">function</span>(i) {</span>
<span id="cb102-16"><a href="#cb102-16" tabindex="-1"></a>  gene_symbol <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gene_expr_matrix)[i]</span>
<span id="cb102-17"><a href="#cb102-17" tabindex="-1"></a>  </span>
<span id="cb102-18"><a href="#cb102-18" tabindex="-1"></a>  covid_expr <span class="ot">&lt;-</span> gene_expr_matrix[i, covid_samples]</span>
<span id="cb102-19"><a href="#cb102-19" tabindex="-1"></a>  control_expr <span class="ot">&lt;-</span> gene_expr_matrix[i, control_samples]</span>
<span id="cb102-20"><a href="#cb102-20" tabindex="-1"></a>  </span>
<span id="cb102-21"><a href="#cb102-21" tabindex="-1"></a>  t_test <span class="ot">&lt;-</span> <span class="fu">t.test</span>(covid_expr, control_expr)</span>
<span id="cb102-22"><a href="#cb102-22" tabindex="-1"></a>  </span>
<span id="cb102-23"><a href="#cb102-23" tabindex="-1"></a>  <span class="fu">tibble</span>(</span>
<span id="cb102-24"><a href="#cb102-24" tabindex="-1"></a>    <span class="at">Gene =</span> gene_symbol,</span>
<span id="cb102-25"><a href="#cb102-25" tabindex="-1"></a>    <span class="at">COVID_mean =</span> <span class="fu">mean</span>(covid_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb102-26"><a href="#cb102-26" tabindex="-1"></a>    <span class="at">Control_mean =</span> <span class="fu">mean</span>(control_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb102-27"><a href="#cb102-27" tabindex="-1"></a>    <span class="at">Fold_change =</span> <span class="fu">mean</span>(covid_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">-</span> <span class="fu">mean</span>(control_expr, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb102-28"><a href="#cb102-28" tabindex="-1"></a>    <span class="at">t_statistic =</span> t_test<span class="sc">$</span>statistic,</span>
<span id="cb102-29"><a href="#cb102-29" tabindex="-1"></a>    <span class="at">p_value =</span> t_test<span class="sc">$</span>p.value</span>
<span id="cb102-30"><a href="#cb102-30" tabindex="-1"></a>  )</span>
<span id="cb102-31"><a href="#cb102-31" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb102-32"><a href="#cb102-32" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb102-33"><a href="#cb102-33" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb102-34"><a href="#cb102-34" tabindex="-1"></a>    <span class="at">p_adjusted =</span> <span class="fu">p.adjust</span>(p_value, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>),</span>
<span id="cb102-35"><a href="#cb102-35" tabindex="-1"></a>    <span class="at">significance =</span> <span class="fu">case_when</span>(</span>
<span id="cb102-36"><a href="#cb102-36" tabindex="-1"></a>      p_adjusted <span class="sc">&lt;</span> <span class="fl">0.001</span> <span class="sc">~</span> <span class="st">&quot;***&quot;</span>,</span>
<span id="cb102-37"><a href="#cb102-37" tabindex="-1"></a>      p_adjusted <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">&quot;**&quot;</span>,</span>
<span id="cb102-38"><a href="#cb102-38" tabindex="-1"></a>      p_adjusted <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">&quot;*&quot;</span>,</span>
<span id="cb102-39"><a href="#cb102-39" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;ns&quot;</span></span>
<span id="cb102-40"><a href="#cb102-40" tabindex="-1"></a>    )</span>
<span id="cb102-41"><a href="#cb102-41" tabindex="-1"></a>  )</span>
<span id="cb102-42"><a href="#cb102-42" tabindex="-1"></a></span>
<span id="cb102-43"><a href="#cb102-43" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">**Statistical Comparison: COVID-19/SARS-CoV-2 vs. Controls**</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## **Statistical Comparison: COVID-19/SARS-CoV-2 vs. Controls**</code></pre>
<div class="sourceCode" id="cb104"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  COVID samples: %d</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">length</span>(covid_samples)))</span></code></pre></div>
<pre><code>##   COVID samples: 12</code></pre>
<div class="sourceCode" id="cb106"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  Control samples: %d</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">length</span>(control_samples)))</span></code></pre></div>
<pre><code>##   Control samples: 28</code></pre>
<div class="sourceCode" id="cb108"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" tabindex="-1"></a><span class="fu">kable</span>(test_results <span class="sc">%&gt;%</span> </span>
<span id="cb108-2"><a href="#cb108-2" tabindex="-1"></a>        <span class="fu">select</span>(Gene, COVID_mean, Control_mean, Fold_change, </span>
<span id="cb108-3"><a href="#cb108-3" tabindex="-1"></a>               p_value, p_adjusted, significance),</span>
<span id="cb108-4"><a href="#cb108-4" tabindex="-1"></a>      <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;COVID Mean&quot;</span>, <span class="st">&quot;Control Mean&quot;</span>, <span class="st">&quot;Δ Expression&quot;</span>, </span>
<span id="cb108-5"><a href="#cb108-5" tabindex="-1"></a>                    <span class="st">&quot;P-value&quot;</span>, <span class="st">&quot;Adj. P-value&quot;</span>, <span class="st">&quot;Sig.&quot;</span>),</span>
<span id="cb108-6"><a href="#cb108-6" tabindex="-1"></a>      <span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb108-7"><a href="#cb108-7" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>))</span></code></pre></div>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
COVID Mean
</th>
<th style="text-align:right;">
Control Mean
</th>
<th style="text-align:right;">
Δ Expression
</th>
<th style="text-align:right;">
P-value
</th>
<th style="text-align:right;">
Adj. P-value
</th>
<th style="text-align:left;">
Sig.
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IRAK2
</td>
<td style="text-align:right;">
4.444
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
1.387
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:left;">
**
</td>
</tr>
<tr>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:right;">
4.529
</td>
<td style="text-align:right;">
3.305
</td>
<td style="text-align:right;">
1.224
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:left;">
***
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8L1
</td>
<td style="text-align:right;">
4.322
</td>
<td style="text-align:right;">
4.196
</td>
<td style="text-align:right;">
0.126
</td>
<td style="text-align:right;">
0.874
</td>
<td style="text-align:right;">
0.874
</td>
<td style="text-align:left;">
ns
</td>
</tr>
<tr>
<td style="text-align:left;">
ANXA8
</td>
<td style="text-align:right;">
3.412
</td>
<td style="text-align:right;">
3.199
</td>
<td style="text-align:right;">
0.213
</td>
<td style="text-align:right;">
0.771
</td>
<td style="text-align:right;">
0.874
</td>
<td style="text-align:left;">
ns
</td>
</tr>
</tbody>
</table>
<p>To assess whether the selected genes show statistically significant
differences between COVID-19 samples and controls, we performed
two-sample t-tests for each gene.</p>
<p>We compared the mean expression in infected versus control samples,
adjusting for multiple testing using the Benjamini-Hochberg procedure to
control the false discovery rate (FDR).</p>
<p>For each gene, we report the mean expression in each condition, the
fold-change, the t-statistic, and both the raw and adjusted p-values.
Significance is annotated using standard thresholds (<em><strong>p &lt;
0.001, </strong>p &lt; 0.01, </em>p &lt; 0.05, ns = not significant).
This analysis quantifies the degree of differential expression and
confirms which genes exhibit consistent upregulation or downregulation
in response to SARS-CoV-2 infection.</p>
</div>
<div id="expression-across-cell-types" class="section level3">
<h3>Expression Across Cell Types</h3>
<div class="sourceCode" id="cb109"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" tabindex="-1"></a><span class="co"># Violin plot by cell type</span></span>
<span id="cb109-2"><a href="#cb109-2" tabindex="-1"></a><span class="fu">ggplot</span>(gene_expr_data, <span class="fu">aes</span>(<span class="at">x =</span> cell_type, <span class="at">y =</span> expression, <span class="at">fill =</span> cell_type)) <span class="sc">+</span></span>
<span id="cb109-3"><a href="#cb109-3" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb109-4"><a href="#cb109-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">outlier.alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb109-5"><a href="#cb109-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Name, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb109-6"><a href="#cb109-6" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb109-7"><a href="#cb109-7" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb109-8"><a href="#cb109-8" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb109-9"><a href="#cb109-9" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb109-10"><a href="#cb109-10" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb109-11"><a href="#cb109-11" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb109-12"><a href="#cb109-12" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Gene Expression Across Cell Types&quot;</span>,</span>
<span id="cb109-13"><a href="#cb109-13" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Cell Type&quot;</span>,</span>
<span id="cb109-14"><a href="#cb109-14" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;log₂(FPKM + 1)&quot;</span></span>
<span id="cb109-15"><a href="#cb109-15" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="covid_files/figure-html/gene_celltype-1.png" width="1152" /></p>
<p>Differences in height reveal cell-type-specific expression because
infection may increase expression in some cell types more than others.
Wide distributions show heterogeneous responses.</p>
<p>We see that the magnitude of expression changes varies by tissue,
meaning the COVID-19 response is highly cell-type-dependent. This
confirms that cell identity is the strongest driver of expression
variation in this dataset.</p>
</div>
<div id="gene-function-summary" class="section level3">
<h3>Gene Function Summary</h3>
<div class="sourceCode" id="cb110"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">**Summary of Selected Genes:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## 
## **Summary of Selected Genes:**</code></pre>
<div class="sourceCode" id="cb112"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(target_genes)) {</span>
<span id="cb112-2"><a href="#cb112-2" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;**%s (%s)**</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb112-3"><a href="#cb112-3" tabindex="-1"></a>              target_genes<span class="sc">$</span>Symbol[i], </span>
<span id="cb112-4"><a href="#cb112-4" tabindex="-1"></a>              target_genes<span class="sc">$</span>GENEID[i]))</span>
<span id="cb112-5"><a href="#cb112-5" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  • log₂FC: %.2f</span><span class="sc">\n</span><span class="st">&quot;</span>, target_genes<span class="sc">$</span>log2FoldChange[i]))</span>
<span id="cb112-6"><a href="#cb112-6" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  • Adjusted P-value: %.2e</span><span class="sc">\n</span><span class="st">&quot;</span>, target_genes<span class="sc">$</span>padj[i]))</span>
<span id="cb112-7"><a href="#cb112-7" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  • Direction: %s in COVID-19</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb112-8"><a href="#cb112-8" tabindex="-1"></a>              <span class="fu">ifelse</span>(target_genes<span class="sc">$</span>log2FoldChange[i] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;Upregulated&quot;</span>, <span class="st">&quot;Downregulated&quot;</span>)))</span>
<span id="cb112-9"><a href="#cb112-9" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;  • Description: %s</span><span class="sc">\n\n</span><span class="st">&quot;</span>, target_genes<span class="sc">$</span>Description[i]))</span>
<span id="cb112-10"><a href="#cb112-10" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## **IRAK2 (IRAK2)**
##   • log₂FC: 1.81
##   • Adjusted P-value: 4.50e-19
##   • Direction: Upregulated in COVID-19
##   • Description: NA
## 
## **SOD2 (SOD2)**
##   • log₂FC: 1.45
##   • Adjusted P-value: 4.65e-19
##   • Direction: Upregulated in COVID-19
##   • Description: NA
## 
## **ANXA8L1 (ANXA8L1)**
##   • log₂FC: -25.37
##   • Adjusted P-value: 2.83e-13
##   • Direction: Downregulated in COVID-19
##   • Description: NA
## 
## **ANXA8 (ANXA8)**
##   • log₂FC: -24.51
##   • Adjusted P-value: 2.11e-12
##   • Direction: Downregulated in COVID-19
##   • Description: NA</code></pre>
<div class="sourceCode" id="cb114"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb114-2"><a href="#cb114-2" tabindex="-1"></a><span class="co"># write_csv(test_results, &quot;targeted_gene_statistics.csv&quot;)</span></span>
<span id="cb114-3"><a href="#cb114-3" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;✓ Statistical results exported to &#39;targeted_gene_statistics.csv&#39;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## ✓ Statistical results exported to &#39;targeted_gene_statistics.csv&#39;</code></pre>
</div>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>Our transcriptomic analysis reveals that SARS-CoV-2 infection induces
distinct and robust changes in host gene expression. <strong>Key
findings include</strong>:</p>
<ul>
<li>Upregulation of inflammatory and interferon-stimulated genes (e.g.,
IRAK2, SOD2), reflecting an activated antiviral and immune
response.</li>
<li>Downregulation of specific epithelial and regulatory genes (e.g.,
ANXA8, ANXA8L1), suggesting tissue-specific dysregulation.</li>
<li>Cell-type-specific expression patterns, with NHBE and A549 cells
showing pronounced transcriptional shifts, highlighting the importance
of cellular context.</li>
<li>Strong co-expression and correlation patterns among selected genes,
indicating shared regulatory pathways during infection.</li>
</ul>
</div>
<div
id="these-results-give-a-molecular-signature-of-covid-19-and-identify-potential-targets-for-further-functional-studies-or-therapeutic-intervention.-overall-the-analysis-demonstrates-how-sars-cov-2-reshapes-the-host-transcriptome-balancing-antiviral-defense-with-inflammatory-damage."
class="section level2">
<h2>These results give a molecular signature of COVID-19 and identify
potential targets for further functional studies or therapeutic
intervention. Overall, the analysis demonstrates how SARS-CoV-2 reshapes
the host transcriptome, balancing antiviral defense with inflammatory
damage.</h2>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Blanco‑Melo, D. et al. <em>Imbalanced Host Response to
SARS-CoV-2 Drives Development of COVID-19.</em> Cell, 2020. <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7227586/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7227586/</a><a
href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Ibid. Upregulation of chemokines (CCL2, CXCL9) and
cytokines (IL‑6) highlights the inflammatory signature in COVID‑19.<a
href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
